{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "day3_day1&2_review.ipynb",
      "version": "0.3.2",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "metadata": {
        "id": "clx9wOTRfWni",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "# Mount your own G Drive on Virtual Machine\n",
        "from google.colab import drive"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "WlfQq1R0fv2c",
        "colab_type": "code",
        "outputId": "648cb0fd-6bac-4cd2-de43-7a5be4ad4439",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 125
        }
      },
      "cell_type": "code",
      "source": [
        "drive.mount('/content/gdrive')"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code\n",
            "\n",
            "Enter your authorization code:\n",
            "··········\n",
            "Mounted at /content/gdrive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "_G35THDkfyuT",
        "colab_type": "code",
        "outputId": "c7825854-8280-47d2-8ff1-828567f9c6ca",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "cd gdrive/My\\ Drive/hbi2019"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/gdrive/My Drive/hbi2019\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "ZvQhzLqqgINb",
        "colab_type": "code",
        "outputId": "87b840b2-cf6a-4b9f-d026-859a1e3374e0",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 6725
        }
      },
      "cell_type": "code",
      "source": [
        "# sra-toolkit installing error가 생길때 해결하기위해 사용\n",
        "! apt-get update \n",
        "# Installing basic genome analysis tools\n",
        "! apt install bwa\n",
        "! apt install samtools\n",
        "! apt install bcftools\n",
        "! apt install sra-toolkit"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\r0% [Working]\r            \rIgn:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease\n",
            "\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (91.18\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease [3,609 B]\n",
            "\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (91.18\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (91.18\r0% [2 InRelease gpgv 3,609 B] [Connecting to archive.ubuntu.com (91.189.88.162)\r                                                                               \rIgn:3 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n",
            "Get:4 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release [564 B]\n",
            "Hit:5 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n",
            "Get:6 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release.gpg [819 B]\n",
            "Get:7 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n",
            "Get:8 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease [21.3 kB]\n",
            "Hit:9 http://archive.ubuntu.com/ubuntu bionic InRelease\n",
            "Get:10 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ Packages [58.4 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n",
            "Get:13 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic InRelease [15.4 kB]\n",
            "Get:14 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [71.4 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]\n",
            "Get:16 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic/main amd64 Packages [31.0 kB]\n",
            "Get:17 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [361 kB]\n",
            "Get:18 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic/main Sources [1,627 kB]\n",
            "Get:19 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [3,910 B]\n",
            "Get:20 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [159 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [725 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [6,968 B]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [953 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [3,659 B]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages [942 B]\n",
            "Get:26 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic/main amd64 Packages [782 kB]\n",
            "Fetched 5,076 kB in 6s (904 kB/s)\n",
            "Reading package lists... Done\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-410\n",
            "Use 'apt autoremove' to remove it.\n",
            "Suggested packages:\n",
            "  samtools\n",
            "The following NEW packages will be installed:\n",
            "  bwa\n",
            "0 upgraded, 1 newly installed, 0 to remove and 127 not upgraded.\n",
            "Need to get 176 kB of archives.\n",
            "After this operation, 404 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bwa amd64 0.7.17-1 [176 kB]\n",
            "Fetched 176 kB in 1s (215 kB/s)\n",
            "Selecting previously unselected package bwa.\n",
            "(Reading database ... 131323 files and directories currently installed.)\n",
            "Preparing to unpack .../bwa_0.7.17-1_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-1) ...\n",
            "Setting up bwa (0.7.17-1) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-410\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libhttp-parser2.7.1 libjs-bootstrap nodejs\n",
            "  nodejs-doc python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  python-cryptography-doc python-cryptography-vectors python-enum34-doc\n",
            "  python-genshi python-lockfile-doc python-openssl-doc python-openssl-dbg\n",
            "  python-setuptools python-pyparsing-doc python-rdflib-doc python-rdflib-tools\n",
            "  python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libhttp-parser2.7.1 libjs-bootstrap nodejs\n",
            "  nodejs-doc python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 127 not upgraded.\n",
            "Need to get 8,533 kB of archives.\n",
            "After this operation, 40.8 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-urllib3 all 1.22-1 [85.1 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.2 [221 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/main amd64 libc-ares2 amd64 1.14.0-1 [37.1 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-parser2.7.1 amd64 2.7.1-2 [20.6 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 8,533 kB in 2s (3,793 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 131340 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.2_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.2) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../33-libc-ares2_1.14.0-1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1) ...\n",
            "Selecting previously unselected package libhttp-parser2.7.1:amd64.\n",
            "Preparing to unpack .../34-libhttp-parser2.7.1_2.7.1-2_amd64.deb ...\n",
            "Unpacking libhttp-parser2.7.1:amd64 (2.7.1-2) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up libhttp-parser2.7.1:amd64 (2.7.1-2) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.2) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-410\n",
            "Use 'apt autoremove' to remove it.\n",
            "Suggested packages:\n",
            "  python-matplotlib texlive-latex-recommended\n",
            "The following NEW packages will be installed:\n",
            "  bcftools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 127 not upgraded.\n",
            "Need to get 503 kB of archives.\n",
            "After this operation, 1,671 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bcftools amd64 1.7-2 [503 kB]\n",
            "Fetched 503 kB in 1s (481 kB/s)\n",
            "Selecting previously unselected package bcftools.\n",
            "(Reading database ... 132784 files and directories currently installed.)\n",
            "Preparing to unpack .../bcftools_1.7-2_amd64.deb ...\n",
            "Unpacking bcftools (1.7-2) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Setting up bcftools (1.7-2) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-410\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libmagic-mgc libmagic1 libmbedcrypto1 libmbedtls-dev\n",
            "  libmbedtls10 libmbedx509-0 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc file libmbedtls-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libmagic-mgc libmagic1 libmbedcrypto1 libmbedtls-dev\n",
            "  libmbedtls10 libmbedx509-0 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "  sra-toolkit\n",
            "0 upgraded, 13 newly installed, 0 to remove and 127 not upgraded.\n",
            "Need to get 4,458 kB of archives.\n",
            "After this operation, 24.8 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic-mgc amd64 1:5.32-2ubuntu0.2 [184 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic1 amd64 1:5.32-2ubuntu0.2 [68.5 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 menu amd64 2.1.47ubuntu2.1 [349 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/universe amd64 blends-common all 0.6.100ubuntu2 [15.0 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedcrypto1 amd64 2.8.0-1 [140 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedx509-0 amd64 2.8.0-1 [34.0 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedtls10 amd64 2.8.0-1 [66.8 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedtls-dev amd64 2.8.0-1 [382 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libkdf5-2 amd64 2.8.2-2+dfsg-1build1 [13.1 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-vdb2 amd64 2.8.2-2+dfsg-1build1 [1,098 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-wvdb2 amd64 2.8.2-2+dfsg-1build1 [997 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 sra-toolkit amd64 2.8.2-5+dfsg-1 [1,102 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/universe amd64 med-config all 3.0.1ubuntu1 [9,404 B]\n",
            "Fetched 4,458 kB in 2s (2,616 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package libmagic-mgc.\n",
            "(Reading database ... 132825 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libmagic-mgc_1%3a5.32-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libmagic-mgc (1:5.32-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libmagic1:amd64.\n",
            "Preparing to unpack .../01-libmagic1_1%3a5.32-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libmagic1:amd64 (1:5.32-2ubuntu0.2) ...\n",
            "Selecting previously unselected package menu.\n",
            "Preparing to unpack .../02-menu_2.1.47ubuntu2.1_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu2.1) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../03-blends-common_0.6.100ubuntu2_all.deb ...\n",
            "Unpacking blends-common (0.6.100ubuntu2) ...\n",
            "Selecting previously unselected package libmbedcrypto1:amd64.\n",
            "Preparing to unpack .../04-libmbedcrypto1_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedcrypto1:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libmbedx509-0:amd64.\n",
            "Preparing to unpack .../05-libmbedx509-0_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedx509-0:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libmbedtls10:amd64.\n",
            "Preparing to unpack .../06-libmbedtls10_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedtls10:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libmbedtls-dev:amd64.\n",
            "Preparing to unpack .../07-libmbedtls-dev_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedtls-dev:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../08-libkdf5-2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../09-libncbi-vdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../10-libncbi-wvdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../11-sra-toolkit_2.8.2-5+dfsg-1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.8.2-5+dfsg-1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../12-med-config_3.0.1ubuntu1_all.deb ...\n",
            "Unpacking med-config (3.0.1ubuntu1) ...\n",
            "Setting up libmbedcrypto1:amd64 (2.8.0-1) ...\n",
            "Setting up menu (2.1.47ubuntu2.1) ...\n",
            "Setting up libmagic-mgc (1:5.32-2ubuntu0.2) ...\n",
            "Setting up libmagic1:amd64 (1:5.32-2ubuntu0.2) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1) ...\n",
            "Setting up libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Setting up libmbedx509-0:amd64 (2.8.0-1) ...\n",
            "Setting up libmbedtls10:amd64 (2.8.0-1) ...\n",
            "Setting up blends-common (0.6.100ubuntu2) ...\n",
            "Setting up libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Setting up libmbedtls-dev:amd64 (2.8.0-1) ...\n",
            "Setting up med-config (3.0.1ubuntu1) ...\n",
            "Adding group `med' (GID 109) ...\n",
            "Done.\n",
            "Setting up libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n",
            "Setting up sra-toolkit (2.8.2-5+dfsg-1) ...\n",
            "Processing triggers for menu (2.1.47ubuntu2.1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1) ...\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "jFjgyiQHgKjX",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "# NCBI SRA info upload\n",
        "# NCBI SRA 홈페이지에 접속, E.coli로 검색, 오른쪽 organism에 Escherichia coli 선택, 왼쪽에 DNA 선택, 자료에 들어가 sample data들이 있는 페이지까지 접속\n",
        "# 저번 시간에 사용했던 Study ID 'SRP071789'를 바로 검색하여도 됨\n",
        "# Download 부분의 RunInfo Table을 눌러 컴퓨터에 sample의 모든 data를 다운\n",
        "# hbi2019 directory내에 ecoli라는 이름의 directory를 새로 만들고 다운받았던 SraRunTable.txt를 ecoli directory에 upload해준다\n",
        "# Upload한 파일의 이름을 Study ID인 SRP071789.txt로 바꾸어준다."
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "8lJhEAPTiVS2",
        "colab_type": "code",
        "outputId": "d5f937d1-a6fa-4f0d-ecc9-b32a41059b83",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "cd ecoli"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/gdrive/My Drive/hbi2019/ecoli\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "LNX_pgO5i24I",
        "colab_type": "code",
        "outputId": "02f6d7e6-bb65-4180-c1d7-34d214285dd1",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "ls"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "SRP071789.txt\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "qFuATqp7i8Qw",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "import pandas as pd"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "5MXZ915BjB3T",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "df_sra = pd.read_csv('./SRP071789.txt', sep = '\\t')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "eujNOSpKjGD7",
        "colab_type": "code",
        "outputId": "f4b792aa-e00c-4928-fab2-36cfa84026ea",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 395
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra.head()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>AvgSpotLen</th>\n",
              "      <th>BioSample</th>\n",
              "      <th>BioSampleModel</th>\n",
              "      <th>Center_Name</th>\n",
              "      <th>DATASTORE_filetype</th>\n",
              "      <th>DATASTORE_provider</th>\n",
              "      <th>DATASTORE_region</th>\n",
              "      <th>Experiment</th>\n",
              "      <th>Instrument</th>\n",
              "      <th>LibraryLayout</th>\n",
              "      <th>...</th>\n",
              "      <th>potential_contaminant</th>\n",
              "      <th>sequence_type</th>\n",
              "      <th>serovar</th>\n",
              "      <th>strain</th>\n",
              "      <th>Assay_Type</th>\n",
              "      <th>BioProject</th>\n",
              "      <th>Consent</th>\n",
              "      <th>InsertSize</th>\n",
              "      <th>LibrarySource</th>\n",
              "      <th>SRA_Study</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>225</td>\n",
              "      <td>SAMN03491946</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>sra</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX999960</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>NaN</td>\n",
              "      <td>O157</td>\n",
              "      <td>H113240287</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN04554948</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1632176</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26:H11</td>\n",
              "      <td>224098</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567806</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643913</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>1819</td>\n",
              "      <td>O166</td>\n",
              "      <td>93311</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567828</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643921</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>10</td>\n",
              "      <td>O113</td>\n",
              "      <td>101270</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567829</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643922</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>17</td>\n",
              "      <td>O103</td>\n",
              "      <td>65974</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 39 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "   AvgSpotLen     BioSample BioSampleModel Center_Name  \\\n",
              "0         225  SAMN03491946    Pathogen.cl         PHE   \n",
              "1         200  SAMN04554948    Pathogen.cl         PHE   \n",
              "2         196  SAMN04567806    Pathogen.cl         PHE   \n",
              "3         197  SAMN04567828    Pathogen.cl         PHE   \n",
              "4         197  SAMN04567829    Pathogen.cl         PHE   \n",
              "\n",
              "                             DATASTORE_filetype DATASTORE_provider  \\\n",
              "0                                           sra               ncbi   \n",
              "1  minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "2  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "3  minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "4  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "\n",
              "  DATASTORE_region  Experiment           Instrument LibraryLayout    ...      \\\n",
              "0              NaN   SRX999960  Illumina HiSeq 2500        PAIRED    ...       \n",
              "1   sra-sos.public  SRX1632176  Illumina HiSeq 2500        PAIRED    ...       \n",
              "2              NaN  SRX1643913  Illumina HiSeq 2500        PAIRED    ...       \n",
              "3   sra-sos.public  SRX1643921  Illumina HiSeq 2500        PAIRED    ...       \n",
              "4              NaN  SRX1643922  Illumina HiSeq 2500        PAIRED    ...       \n",
              "\n",
              "  potential_contaminant sequence_type  serovar      strain  Assay_Type  \\\n",
              "0           not_checked           NaN     O157  H113240287         WGS   \n",
              "1           not_checked            21  O26:H11      224098         WGS   \n",
              "2           not_checked          1819     O166       93311         WGS   \n",
              "3           not_checked            10     O113      101270         WGS   \n",
              "4           not_checked            17     O103       65974         WGS   \n",
              "\n",
              "    BioProject Consent InsertSize LibrarySource  SRA_Study  \n",
              "0  PRJNA315192  public          0       GENOMIC  SRP071789  \n",
              "1  PRJNA315192  public          0       GENOMIC  SRP071789  \n",
              "2  PRJNA315192  public          0       GENOMIC  SRP071789  \n",
              "3  PRJNA315192  public          0       GENOMIC  SRP071789  \n",
              "4  PRJNA315192  public          0       GENOMIC  SRP071789  \n",
              "\n",
              "[5 rows x 39 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 10
        }
      ]
    },
    {
      "metadata": {
        "id": "N6CDQ5b5jHPK",
        "colab_type": "code",
        "outputId": "cf5965b8-9225-48e2-9cb7-e5155502d720",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 210
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra.columns\n",
        "\n",
        "# header를 통해 다양한 정보가 있음을 알 수 있음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Index(['AvgSpotLen', 'BioSample', 'BioSampleModel', 'Center_Name',\n",
              "       'DATASTORE_filetype', 'DATASTORE_provider', 'DATASTORE_region',\n",
              "       'Experiment', 'Instrument', 'LibraryLayout', 'LibrarySelection',\n",
              "       'Library_Name', 'LoadDate', 'MBases', 'MBytes', 'Organism', 'Platform',\n",
              "       'ReleaseDate', 'Run', 'SRA_Sample', 'Sample_Name', 'collected_by',\n",
              "       'collection_date', 'geo_loc_name', 'host', 'host_disease',\n",
              "       'isolate_name_alias', 'isolation_source', 'lat_lon',\n",
              "       'potential_contaminant', 'sequence_type', 'serovar', 'strain',\n",
              "       'Assay_Type', 'BioProject', 'Consent', 'InsertSize', 'LibrarySource',\n",
              "       'SRA_Study'],\n",
              "      dtype='object')"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 11
        }
      ]
    },
    {
      "metadata": {
        "id": "ZzUIYoG2jOfz",
        "colab_type": "code",
        "outputId": "f8176d3a-4bea-437e-a7b9-e9121722e638",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1108
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra['Organism']"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "0            Escherichia coli\n",
              "1            Escherichia coli\n",
              "2            Escherichia coli\n",
              "3            Escherichia coli\n",
              "4            Escherichia coli\n",
              "5            Escherichia coli\n",
              "6            Escherichia coli\n",
              "7            Escherichia coli\n",
              "8            Escherichia coli\n",
              "9            Escherichia coli\n",
              "10           Escherichia coli\n",
              "11           Escherichia coli\n",
              "12           Escherichia coli\n",
              "13           Escherichia coli\n",
              "14           Escherichia coli\n",
              "15           Escherichia coli\n",
              "16           Escherichia coli\n",
              "17           Escherichia coli\n",
              "18           Escherichia coli\n",
              "19           Escherichia coli\n",
              "20           Escherichia coli\n",
              "21           Escherichia coli\n",
              "22           Escherichia coli\n",
              "23           Escherichia coli\n",
              "24           Escherichia coli\n",
              "25           Escherichia coli\n",
              "26           Escherichia coli\n",
              "27           Escherichia coli\n",
              "28           Escherichia coli\n",
              "29           Escherichia coli\n",
              "                 ...         \n",
              "13145       Shigella flexneri\n",
              "13146         Shigella sonnei\n",
              "13147        Escherichia coli\n",
              "13148        Escherichia coli\n",
              "13149        Escherichia coli\n",
              "13150         Shigella sonnei\n",
              "13151        Escherichia coli\n",
              "13152        Escherichia coli\n",
              "13153        Escherichia coli\n",
              "13154        Escherichia coli\n",
              "13155       Shigella flexneri\n",
              "13156         Shigella sonnei\n",
              "13157       Shigella flexneri\n",
              "13158       Shigella flexneri\n",
              "13159        Escherichia coli\n",
              "13160         Shigella boydii\n",
              "13161       Shigella flexneri\n",
              "13162         Shigella sonnei\n",
              "13163         Shigella sonnei\n",
              "13164         Shigella boydii\n",
              "13165         Shigella sonnei\n",
              "13166       Shigella flexneri\n",
              "13167    Escherichia albertii\n",
              "13168       Shigella flexneri\n",
              "13169         Shigella sonnei\n",
              "13170       Shigella flexneri\n",
              "13171         Shigella sonnei\n",
              "13172       Shigella flexneri\n",
              "13173         Shigella sonnei\n",
              "13174        Escherichia coli\n",
              "Name: Organism, Length: 13175, dtype: object"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 12
        }
      ]
    },
    {
      "metadata": {
        "id": "PWeF2sc2jd4X",
        "colab_type": "code",
        "outputId": "91788e4c-266a-4eb4-dc70-43062ad23f9b",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 404
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra['Organism'].unique()\n",
        "# organism에 있는 종류만 알려줌"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['Escherichia coli', 'Shigella flexneri', 'Shigella sonnei',\n",
              "       'Escherichia albertii', 'Shigella boydii', 'Shigella dysenteriae',\n",
              "       'Escherichia sp. 260038',\n",
              "       'Salmonella enterica subsp. enterica serovar Braenderup',\n",
              "       'Listeria monocytogenes', 'Shigella sp. 262643',\n",
              "       'Salmonella sp. 178897', 'Shigella sp. 254601',\n",
              "       'Shigella sp. 176296', 'Shigella sp. 312160',\n",
              "       'Shigella sp. 312103',\n",
              "       'Salmonella enterica subsp. enterica serovar Enteritidis',\n",
              "       'Escherichia sp. 126125', 'Shigella sp. 183737',\n",
              "       'Shigella sp. 184399', 'Shigella sp. 200615',\n",
              "       'Shigella sp. 255528', 'Shigella sp. 283097',\n",
              "       'Shigella sp. 171147', 'Shigella sp. 279355',\n",
              "       'Shigella sp. 279345', 'Shigella sp. 296637',\n",
              "       'Shigella sp. 273561', 'bacterium 264140', 'Shigella sp. 175547',\n",
              "       'Shigella sp. 278204', 'Shigella sp. 298909',\n",
              "       'Shigella sp. 188334', 'Shigella sp. 279353',\n",
              "       'Shigella sp. 217069', 'Shigella sp. 168500',\n",
              "       'Shigella sp. 188144', 'Shigella sp. 291764',\n",
              "       'Shigella sp. 170304', 'Shigella sp. 127555',\n",
              "       'Escherichia sp. 127405', 'Escherichia sp. 134258',\n",
              "       'Shigella sp. 255519', 'Escherichia sp. 286788'], dtype=object)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 14
        }
      ]
    },
    {
      "metadata": {
        "id": "fuUlGyeojWyp",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "m1 = df_sra['Organism'] == 'Escherichia coli'"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "BCPsIMWBjcy_",
        "colab_type": "code",
        "outputId": "61f63bd2-a0b9-43eb-9e42-1ccbd74d5ee6",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 3901
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>AvgSpotLen</th>\n",
              "      <th>BioSample</th>\n",
              "      <th>BioSampleModel</th>\n",
              "      <th>Center_Name</th>\n",
              "      <th>DATASTORE_filetype</th>\n",
              "      <th>DATASTORE_provider</th>\n",
              "      <th>DATASTORE_region</th>\n",
              "      <th>Experiment</th>\n",
              "      <th>Instrument</th>\n",
              "      <th>LibraryLayout</th>\n",
              "      <th>...</th>\n",
              "      <th>potential_contaminant</th>\n",
              "      <th>sequence_type</th>\n",
              "      <th>serovar</th>\n",
              "      <th>strain</th>\n",
              "      <th>Assay_Type</th>\n",
              "      <th>BioProject</th>\n",
              "      <th>Consent</th>\n",
              "      <th>InsertSize</th>\n",
              "      <th>LibrarySource</th>\n",
              "      <th>SRA_Study</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>225</td>\n",
              "      <td>SAMN03491946</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>sra</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX999960</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>NaN</td>\n",
              "      <td>O157</td>\n",
              "      <td>H113240287</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN04554948</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1632176</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26:H11</td>\n",
              "      <td>224098</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567806</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643913</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>1819</td>\n",
              "      <td>O166</td>\n",
              "      <td>93311</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567828</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643921</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>10</td>\n",
              "      <td>O113</td>\n",
              "      <td>101270</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567829</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643922</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>17</td>\n",
              "      <td>O103</td>\n",
              "      <td>65974</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567827</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643920</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>25</td>\n",
              "      <td>O128ab</td>\n",
              "      <td>93332</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567826</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643919</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>17</td>\n",
              "      <td>O103</td>\n",
              "      <td>101300</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567825</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643918</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>-</td>\n",
              "      <td>O177</td>\n",
              "      <td>93329</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567824</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643917</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>278</td>\n",
              "      <td>O178</td>\n",
              "      <td>93334</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567823</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643916</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>301</td>\n",
              "      <td>O128ad</td>\n",
              "      <td>101267</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567808</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643915</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26</td>\n",
              "      <td>93310</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>195</td>\n",
              "      <td>SAMN04567807</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643914</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26</td>\n",
              "      <td>93309</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567838</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643944</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>504</td>\n",
              "      <td>Rough</td>\n",
              "      <td>101302</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567837</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643943</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>442</td>\n",
              "      <td>O146</td>\n",
              "      <td>101276</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567836</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643942</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>335</td>\n",
              "      <td>O55</td>\n",
              "      <td>101285</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567835</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643941</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>335</td>\n",
              "      <td>O55</td>\n",
              "      <td>101299</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567840</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643946</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>33</td>\n",
              "      <td>Rough</td>\n",
              "      <td>93261</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN04567970</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643962</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>386</td>\n",
              "      <td>O103</td>\n",
              "      <td>65938</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567969</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643961</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>550</td>\n",
              "      <td>O75</td>\n",
              "      <td>101303</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN04567968</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643947</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>335</td>\n",
              "      <td>O55</td>\n",
              "      <td>66002</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567839</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643945</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>442</td>\n",
              "      <td>O146</td>\n",
              "      <td>101282</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN04567834</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643940</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>17</td>\n",
              "      <td>O103</td>\n",
              "      <td>65928</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567833</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1643939</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>335</td>\n",
              "      <td>O55</td>\n",
              "      <td>70626</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567832</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643938</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>342</td>\n",
              "      <td>O5</td>\n",
              "      <td>93336</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567830</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643936</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>10</td>\n",
              "      <td>O38</td>\n",
              "      <td>93260</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>195</td>\n",
              "      <td>SAMN04567831</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1643937</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26</td>\n",
              "      <td>93304</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN04567977</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1644130</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>335</td>\n",
              "      <td>O55</td>\n",
              "      <td>65982</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567976</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX1644129</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>21</td>\n",
              "      <td>O26</td>\n",
              "      <td>93322</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN04567975</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1644128</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>738</td>\n",
              "      <td>O146</td>\n",
              "      <td>101301</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN04567974</td>\n",
              "      <td>Pathogen.cl</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX1644127</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>73</td>\n",
              "      <td>O120</td>\n",
              "      <td>93269</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13085</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11108591</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5509692</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>738</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>699804</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13087</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11149198</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5527789</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>702965</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13088</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11149199</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5527792</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>702967</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13089</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11149205</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5527794</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>702966</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13090</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11155079</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5530611</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>703189</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13097</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11155078</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5530618</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>703177</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13098</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11155084</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5530619</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>8291</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704748</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13103</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157147</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532770</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>705209</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13104</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157151</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532771</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704779</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13105</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11157150</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532772</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>675</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704780</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13106</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157149</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532773</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>17</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704778</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13107</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157148</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532774</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>33</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704771</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13111</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157153</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532778</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>738</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704772</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13112</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11157160</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5532779</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>404</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>704304</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13130</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11173090</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5544769</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>538</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>699737</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13134</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN11192336</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5557000</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707202</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13135</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11192861</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5557020</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707211</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13137</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11192860</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5557022</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707201</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13139</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11236582</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561491</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>1804</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707200</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13141</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11236588</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561570</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>16</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707072</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13143</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11236673</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561587</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>5292</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707212</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13147</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN11236672</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561592</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>122</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707160</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13148</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN11236675</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561595</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707070</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13149</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN11236676</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561596</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>6</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707203</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13151</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11236671</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5561598</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>1819</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>707071</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13152</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11244099</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5564384</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>17</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>708242</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13153</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11244101</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5564385</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>708236</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13154</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11244098</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5564386</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>1804</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>708177</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13159</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11244102</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5564416</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>17</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>708243</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13174</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN11251984</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5573636</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>56</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>708244</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>7626 rows × 39 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "       AvgSpotLen     BioSample BioSampleModel Center_Name  \\\n",
              "0             225  SAMN03491946    Pathogen.cl         PHE   \n",
              "1             200  SAMN04554948    Pathogen.cl         PHE   \n",
              "2             196  SAMN04567806    Pathogen.cl         PHE   \n",
              "3             197  SAMN04567828    Pathogen.cl         PHE   \n",
              "4             197  SAMN04567829    Pathogen.cl         PHE   \n",
              "5             196  SAMN04567827    Pathogen.cl         PHE   \n",
              "6             196  SAMN04567826    Pathogen.cl         PHE   \n",
              "7             196  SAMN04567825    Pathogen.cl         PHE   \n",
              "8             196  SAMN04567824    Pathogen.cl         PHE   \n",
              "9             197  SAMN04567823    Pathogen.cl         PHE   \n",
              "10            196  SAMN04567808    Pathogen.cl         PHE   \n",
              "11            195  SAMN04567807    Pathogen.cl         PHE   \n",
              "12            196  SAMN04567838    Pathogen.cl         PHE   \n",
              "13            196  SAMN04567837    Pathogen.cl         PHE   \n",
              "14            197  SAMN04567836    Pathogen.cl         PHE   \n",
              "15            197  SAMN04567835    Pathogen.cl         PHE   \n",
              "16            196  SAMN04567840    Pathogen.cl         PHE   \n",
              "17            198  SAMN04567970    Pathogen.cl         PHE   \n",
              "18            197  SAMN04567969    Pathogen.cl         PHE   \n",
              "19            198  SAMN04567968    Pathogen.cl         PHE   \n",
              "20            197  SAMN04567839    Pathogen.cl         PHE   \n",
              "21            198  SAMN04567834    Pathogen.cl         PHE   \n",
              "22            197  SAMN04567833    Pathogen.cl         PHE   \n",
              "23            196  SAMN04567832    Pathogen.cl         PHE   \n",
              "24            196  SAMN04567830    Pathogen.cl         PHE   \n",
              "25            195  SAMN04567831    Pathogen.cl         PHE   \n",
              "26            198  SAMN04567977    Pathogen.cl         PHE   \n",
              "27            196  SAMN04567976    Pathogen.cl         PHE   \n",
              "28            197  SAMN04567975    Pathogen.cl         PHE   \n",
              "29            196  SAMN04567974    Pathogen.cl         PHE   \n",
              "...           ...           ...            ...         ...   \n",
              "13085         200  SAMN11108591   Pathogen.env         PHE   \n",
              "13087         200  SAMN11149198   Pathogen.env         PHE   \n",
              "13088         200  SAMN11149199   Pathogen.env         PHE   \n",
              "13089         200  SAMN11149205   Pathogen.env         PHE   \n",
              "13090         200  SAMN11155079   Pathogen.env         PHE   \n",
              "13097         200  SAMN11155078   Pathogen.env         PHE   \n",
              "13098         200  SAMN11155084   Pathogen.env         PHE   \n",
              "13103         199  SAMN11157147   Pathogen.env         PHE   \n",
              "13104         199  SAMN11157151   Pathogen.env         PHE   \n",
              "13105         200  SAMN11157150   Pathogen.env         PHE   \n",
              "13106         199  SAMN11157149   Pathogen.env         PHE   \n",
              "13107         199  SAMN11157148   Pathogen.env         PHE   \n",
              "13111         199  SAMN11157153   Pathogen.env         PHE   \n",
              "13112         199  SAMN11157160   Pathogen.env         PHE   \n",
              "13130         200  SAMN11173090   Pathogen.env         PHE   \n",
              "13134         197  SAMN11192336   Pathogen.env         PHE   \n",
              "13135         200  SAMN11192861   Pathogen.env         PHE   \n",
              "13137         199  SAMN11192860   Pathogen.env         PHE   \n",
              "13139         200  SAMN11236582   Pathogen.env         PHE   \n",
              "13141         199  SAMN11236588   Pathogen.env         PHE   \n",
              "13143         199  SAMN11236673   Pathogen.env         PHE   \n",
              "13147         198  SAMN11236672   Pathogen.env         PHE   \n",
              "13148         197  SAMN11236675   Pathogen.env         PHE   \n",
              "13149         200  SAMN11236676   Pathogen.env         PHE   \n",
              "13151         199  SAMN11236671   Pathogen.env         PHE   \n",
              "13152         199  SAMN11244099   Pathogen.env         PHE   \n",
              "13153         199  SAMN11244101   Pathogen.env         PHE   \n",
              "13154         199  SAMN11244098   Pathogen.env         PHE   \n",
              "13159         199  SAMN11244102   Pathogen.env         PHE   \n",
              "13174         199  SAMN11251984   Pathogen.env         PHE   \n",
              "\n",
              "                                 DATASTORE_filetype DATASTORE_provider  \\\n",
              "0                                               sra               ncbi   \n",
              "1      minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "2      minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "3      minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "4      minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5      minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "6      minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "7      minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "8      minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "9      minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "11     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "12     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "14     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "15     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "16     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "17     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "18     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "19     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "20     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "21     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "22     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "23     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "24     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "25     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "26     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "27     minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "28     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "29     minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "...                                             ...                ...   \n",
              "13085  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13087  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13088  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13089  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13090  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13097  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13098  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13103  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13104  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13105  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13106  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13107  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13111  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13112  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13130  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13134  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13135  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13137  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13139  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13141  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13143  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13147  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13148  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13149  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13151  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13152  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13153  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13154  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13159  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "13174  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "\n",
              "      DATASTORE_region  Experiment           Instrument LibraryLayout  \\\n",
              "0                  NaN   SRX999960  Illumina HiSeq 2500        PAIRED   \n",
              "1       sra-sos.public  SRX1632176  Illumina HiSeq 2500        PAIRED   \n",
              "2                  NaN  SRX1643913  Illumina HiSeq 2500        PAIRED   \n",
              "3       sra-sos.public  SRX1643921  Illumina HiSeq 2500        PAIRED   \n",
              "4                  NaN  SRX1643922  Illumina HiSeq 2500        PAIRED   \n",
              "5       sra-sos.public  SRX1643920  Illumina HiSeq 2500        PAIRED   \n",
              "6                  NaN  SRX1643919  Illumina HiSeq 2500        PAIRED   \n",
              "7                  NaN  SRX1643918  Illumina HiSeq 2500        PAIRED   \n",
              "8       sra-sos.public  SRX1643917  Illumina HiSeq 2500        PAIRED   \n",
              "9                  NaN  SRX1643916  Illumina HiSeq 2500        PAIRED   \n",
              "10      sra-sos.public  SRX1643915  Illumina HiSeq 2500        PAIRED   \n",
              "11      sra-sos.public  SRX1643914  Illumina HiSeq 2500        PAIRED   \n",
              "12                 NaN  SRX1643944  Illumina HiSeq 2500        PAIRED   \n",
              "13                 NaN  SRX1643943  Illumina HiSeq 2500        PAIRED   \n",
              "14                 NaN  SRX1643942  Illumina HiSeq 2500        PAIRED   \n",
              "15      sra-sos.public  SRX1643941  Illumina HiSeq 2500        PAIRED   \n",
              "16                 NaN  SRX1643946  Illumina HiSeq 2500        PAIRED   \n",
              "17      sra-sos.public  SRX1643962  Illumina HiSeq 2500        PAIRED   \n",
              "18      sra-sos.public  SRX1643961  Illumina HiSeq 2500        PAIRED   \n",
              "19                 NaN  SRX1643947  Illumina HiSeq 2500        PAIRED   \n",
              "20                 NaN  SRX1643945  Illumina HiSeq 2500        PAIRED   \n",
              "21      sra-sos.public  SRX1643940  Illumina HiSeq 2500        PAIRED   \n",
              "22                 NaN  SRX1643939  Illumina HiSeq 2500        PAIRED   \n",
              "23      sra-sos.public  SRX1643938  Illumina HiSeq 2500        PAIRED   \n",
              "24      sra-sos.public  SRX1643936  Illumina HiSeq 2500        PAIRED   \n",
              "25      sra-sos.public  SRX1643937  Illumina HiSeq 2500        PAIRED   \n",
              "26      sra-sos.public  SRX1644130  Illumina HiSeq 2500        PAIRED   \n",
              "27      sra-sos.public  SRX1644129  Illumina HiSeq 2500        PAIRED   \n",
              "28                 NaN  SRX1644128  Illumina HiSeq 2500        PAIRED   \n",
              "29                 NaN  SRX1644127  Illumina HiSeq 2500        PAIRED   \n",
              "...                ...         ...                  ...           ...   \n",
              "13085              NaN  SRX5509692  Illumina HiSeq 2500        PAIRED   \n",
              "13087              NaN  SRX5527789  Illumina HiSeq 2500        PAIRED   \n",
              "13088              NaN  SRX5527792  Illumina HiSeq 2500        PAIRED   \n",
              "13089              NaN  SRX5527794  Illumina HiSeq 2500        PAIRED   \n",
              "13090              NaN  SRX5530611  Illumina HiSeq 2500        PAIRED   \n",
              "13097              NaN  SRX5530618  Illumina HiSeq 2500        PAIRED   \n",
              "13098              NaN  SRX5530619  Illumina HiSeq 2500        PAIRED   \n",
              "13103              NaN  SRX5532770  Illumina HiSeq 2500        PAIRED   \n",
              "13104              NaN  SRX5532771  Illumina HiSeq 2500        PAIRED   \n",
              "13105              NaN  SRX5532772  Illumina HiSeq 2500        PAIRED   \n",
              "13106              NaN  SRX5532773  Illumina HiSeq 2500        PAIRED   \n",
              "13107              NaN  SRX5532774  Illumina HiSeq 2500        PAIRED   \n",
              "13111              NaN  SRX5532778  Illumina HiSeq 2500        PAIRED   \n",
              "13112              NaN  SRX5532779  Illumina HiSeq 2500        PAIRED   \n",
              "13130              NaN  SRX5544769  Illumina HiSeq 2500        PAIRED   \n",
              "13134              NaN  SRX5557000  Illumina HiSeq 2500        PAIRED   \n",
              "13135              NaN  SRX5557020  Illumina HiSeq 2500        PAIRED   \n",
              "13137              NaN  SRX5557022  Illumina HiSeq 2500        PAIRED   \n",
              "13139              NaN  SRX5561491  Illumina HiSeq 2500        PAIRED   \n",
              "13141              NaN  SRX5561570  Illumina HiSeq 2500        PAIRED   \n",
              "13143              NaN  SRX5561587  Illumina HiSeq 2500        PAIRED   \n",
              "13147              NaN  SRX5561592  Illumina HiSeq 2500        PAIRED   \n",
              "13148              NaN  SRX5561595  Illumina HiSeq 2500        PAIRED   \n",
              "13149              NaN  SRX5561596  Illumina HiSeq 2500        PAIRED   \n",
              "13151              NaN  SRX5561598  Illumina HiSeq 2500        PAIRED   \n",
              "13152              NaN  SRX5564384  Illumina HiSeq 2500        PAIRED   \n",
              "13153              NaN  SRX5564385  Illumina HiSeq 2500        PAIRED   \n",
              "13154              NaN  SRX5564386  Illumina HiSeq 2500        PAIRED   \n",
              "13159              NaN  SRX5564416  Illumina HiSeq 2500        PAIRED   \n",
              "13174              NaN  SRX5573636  Illumina HiSeq 2500        PAIRED   \n",
              "\n",
              "         ...     potential_contaminant sequence_type           serovar  \\\n",
              "0        ...               not_checked           NaN              O157   \n",
              "1        ...               not_checked            21           O26:H11   \n",
              "2        ...               not_checked          1819              O166   \n",
              "3        ...               not_checked            10              O113   \n",
              "4        ...               not_checked            17              O103   \n",
              "5        ...               not_checked            25            O128ab   \n",
              "6        ...               not_checked            17              O103   \n",
              "7        ...               not_checked             -              O177   \n",
              "8        ...               not_checked           278              O178   \n",
              "9        ...               not_checked           301            O128ad   \n",
              "10       ...               not_checked            21               O26   \n",
              "11       ...               not_checked            21               O26   \n",
              "12       ...               not_checked           504             Rough   \n",
              "13       ...               not_checked           442              O146   \n",
              "14       ...               not_checked           335               O55   \n",
              "15       ...               not_checked           335               O55   \n",
              "16       ...               not_checked            33             Rough   \n",
              "17       ...               not_checked           386              O103   \n",
              "18       ...               not_checked           550               O75   \n",
              "19       ...               not_checked           335               O55   \n",
              "20       ...               not_checked           442              O146   \n",
              "21       ...               not_checked            17              O103   \n",
              "22       ...               not_checked           335               O55   \n",
              "23       ...               not_checked           342                O5   \n",
              "24       ...               not_checked            10               O38   \n",
              "25       ...               not_checked            21               O26   \n",
              "26       ...               not_checked           335               O55   \n",
              "27       ...               not_checked            21               O26   \n",
              "28       ...               not_checked           738              O146   \n",
              "29       ...               not_checked            73              O120   \n",
              "...      ...                       ...           ...               ...   \n",
              "13085    ...             None detected           738  Escherichia coli   \n",
              "13087    ...             None detected            11  Escherichia coli   \n",
              "13088    ...             None detected            11  Escherichia coli   \n",
              "13089    ...             None detected            11  Escherichia coli   \n",
              "13090    ...             None detected            11  Escherichia coli   \n",
              "13097    ...             None detected            11  Escherichia coli   \n",
              "13098    ...             None detected          8291  Escherichia coli   \n",
              "13103    ...             None detected            11  Escherichia coli   \n",
              "13104    ...             None detected            10  Escherichia coli   \n",
              "13105    ...             None detected           675  Escherichia coli   \n",
              "13106    ...             None detected            17  Escherichia coli   \n",
              "13107    ...             None detected            33  Escherichia coli   \n",
              "13111    ...             None detected           738  Escherichia coli   \n",
              "13112    ...             None detected           404  Escherichia coli   \n",
              "13130    ...             None detected           538  Escherichia coli   \n",
              "13134    ...             None detected            11  Escherichia coli   \n",
              "13135    ...             None detected            11  Escherichia coli   \n",
              "13137    ...             None detected            11  Escherichia coli   \n",
              "13139    ...             None detected          1804  Escherichia coli   \n",
              "13141    ...             None detected            16  Escherichia coli   \n",
              "13143    ...             None detected          5292  Escherichia coli   \n",
              "13147    ...             None detected           122  Escherichia coli   \n",
              "13148    ...             None detected            10  Escherichia coli   \n",
              "13149    ...             None detected             6  Escherichia coli   \n",
              "13151    ...             None detected          1819  Escherichia coli   \n",
              "13152    ...             None detected            17  Escherichia coli   \n",
              "13153    ...             None detected            11  Escherichia coli   \n",
              "13154    ...             None detected          1804  Escherichia coli   \n",
              "13159    ...             None detected            17  Escherichia coli   \n",
              "13174    ...             None detected            56  Escherichia coli   \n",
              "\n",
              "           strain  Assay_Type   BioProject Consent InsertSize LibrarySource  \\\n",
              "0      H113240287         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "1          224098         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "2           93311         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "3          101270         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "4           65974         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5           93332         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6          101300         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "7           93329         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "8           93334         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "9          101267         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10          93310         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11          93309         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12         101302         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13         101276         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "14         101285         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "15         101299         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "16          93261         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "17          65938         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "18         101303         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "19          66002         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "20         101282         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "21          65928         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "22          70626         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "23          93336         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "24          93260         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "25          93304         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "26          65982         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "27          93322         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "28         101301         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "29          93269         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "...           ...         ...          ...     ...        ...           ...   \n",
              "13085      699804         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13087      702965         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13088      702967         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13089      702966         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13090      703189         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13097      703177         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13098      704748         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13103      705209         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13104      704779         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13105      704780         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13106      704778         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13107      704771         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13111      704772         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13112      704304         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13130      699737         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13134      707202         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13135      707211         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13137      707201         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13139      707200         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13141      707072         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13143      707212         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13147      707160         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13148      707070         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13149      707203         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13151      707071         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13152      708242         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13153      708236         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13154      708177         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13159      708243         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "13174      708244         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "\n",
              "       SRA_Study  \n",
              "0      SRP071789  \n",
              "1      SRP071789  \n",
              "2      SRP071789  \n",
              "3      SRP071789  \n",
              "4      SRP071789  \n",
              "5      SRP071789  \n",
              "6      SRP071789  \n",
              "7      SRP071789  \n",
              "8      SRP071789  \n",
              "9      SRP071789  \n",
              "10     SRP071789  \n",
              "11     SRP071789  \n",
              "12     SRP071789  \n",
              "13     SRP071789  \n",
              "14     SRP071789  \n",
              "15     SRP071789  \n",
              "16     SRP071789  \n",
              "17     SRP071789  \n",
              "18     SRP071789  \n",
              "19     SRP071789  \n",
              "20     SRP071789  \n",
              "21     SRP071789  \n",
              "22     SRP071789  \n",
              "23     SRP071789  \n",
              "24     SRP071789  \n",
              "25     SRP071789  \n",
              "26     SRP071789  \n",
              "27     SRP071789  \n",
              "28     SRP071789  \n",
              "29     SRP071789  \n",
              "...          ...  \n",
              "13085  SRP071789  \n",
              "13087  SRP071789  \n",
              "13088  SRP071789  \n",
              "13089  SRP071789  \n",
              "13090  SRP071789  \n",
              "13097  SRP071789  \n",
              "13098  SRP071789  \n",
              "13103  SRP071789  \n",
              "13104  SRP071789  \n",
              "13105  SRP071789  \n",
              "13106  SRP071789  \n",
              "13107  SRP071789  \n",
              "13111  SRP071789  \n",
              "13112  SRP071789  \n",
              "13130  SRP071789  \n",
              "13134  SRP071789  \n",
              "13135  SRP071789  \n",
              "13137  SRP071789  \n",
              "13139  SRP071789  \n",
              "13141  SRP071789  \n",
              "13143  SRP071789  \n",
              "13147  SRP071789  \n",
              "13148  SRP071789  \n",
              "13149  SRP071789  \n",
              "13151  SRP071789  \n",
              "13152  SRP071789  \n",
              "13153  SRP071789  \n",
              "13154  SRP071789  \n",
              "13159  SRP071789  \n",
              "13174  SRP071789  \n",
              "\n",
              "[7626 rows x 39 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 15
        }
      ]
    },
    {
      "metadata": {
        "id": "CF9zBR3Yjr4u",
        "colab_type": "code",
        "outputId": "7304cbf8-1e56-400d-9ac5-7923007e7a3a",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1108
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1]['Organism']\n",
        "# masking했기때문에 Escherichia coli만 있는 것을 알 수 있음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "0        Escherichia coli\n",
              "1        Escherichia coli\n",
              "2        Escherichia coli\n",
              "3        Escherichia coli\n",
              "4        Escherichia coli\n",
              "5        Escherichia coli\n",
              "6        Escherichia coli\n",
              "7        Escherichia coli\n",
              "8        Escherichia coli\n",
              "9        Escherichia coli\n",
              "10       Escherichia coli\n",
              "11       Escherichia coli\n",
              "12       Escherichia coli\n",
              "13       Escherichia coli\n",
              "14       Escherichia coli\n",
              "15       Escherichia coli\n",
              "16       Escherichia coli\n",
              "17       Escherichia coli\n",
              "18       Escherichia coli\n",
              "19       Escherichia coli\n",
              "20       Escherichia coli\n",
              "21       Escherichia coli\n",
              "22       Escherichia coli\n",
              "23       Escherichia coli\n",
              "24       Escherichia coli\n",
              "25       Escherichia coli\n",
              "26       Escherichia coli\n",
              "27       Escherichia coli\n",
              "28       Escherichia coli\n",
              "29       Escherichia coli\n",
              "               ...       \n",
              "13085    Escherichia coli\n",
              "13087    Escherichia coli\n",
              "13088    Escherichia coli\n",
              "13089    Escherichia coli\n",
              "13090    Escherichia coli\n",
              "13097    Escherichia coli\n",
              "13098    Escherichia coli\n",
              "13103    Escherichia coli\n",
              "13104    Escherichia coli\n",
              "13105    Escherichia coli\n",
              "13106    Escherichia coli\n",
              "13107    Escherichia coli\n",
              "13111    Escherichia coli\n",
              "13112    Escherichia coli\n",
              "13130    Escherichia coli\n",
              "13134    Escherichia coli\n",
              "13135    Escherichia coli\n",
              "13137    Escherichia coli\n",
              "13139    Escherichia coli\n",
              "13141    Escherichia coli\n",
              "13143    Escherichia coli\n",
              "13147    Escherichia coli\n",
              "13148    Escherichia coli\n",
              "13149    Escherichia coli\n",
              "13151    Escherichia coli\n",
              "13152    Escherichia coli\n",
              "13153    Escherichia coli\n",
              "13154    Escherichia coli\n",
              "13159    Escherichia coli\n",
              "13174    Escherichia coli\n",
              "Name: Organism, Length: 7626, dtype: object"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 16
        }
      ]
    },
    {
      "metadata": {
        "id": "tlWC25dZjwYz",
        "colab_type": "code",
        "outputId": "d3154bf9-aae2-4f95-f648-08279bea5056",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1]['Organism'].unique()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['Escherichia coli'], dtype=object)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 17
        }
      ]
    },
    {
      "metadata": {
        "id": "WLTJ4Bobj9jA",
        "colab_type": "code",
        "outputId": "9cce092d-fcde-47fe-9e75-0f6dc244c230",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 70
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra['isolation_source'].unique()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['Human', 'Other', 'Animal', 'Enviromental', 'Food', 'human',\n",
              "       'food', 'animal', 'missing', 'None', 'water', 'feces',\n",
              "       'environmental', 'other'], dtype=object)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 18
        }
      ]
    },
    {
      "metadata": {
        "id": "XC4H6wTQkE5d",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "m2 = df_sra['isolation_source'] == 'food'"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "LTFvftpRkfF7",
        "colab_type": "code",
        "outputId": "0a06e282-406d-476c-eede-f46faca0936d",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 3918
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1&m2]\n",
        "# 이렇게 masking하는 이유는 임의로 data의 양을 줄이기 위함으로 분석을 시행할 땐 실험자의 기준대로 masking할 필요가 있다"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>AvgSpotLen</th>\n",
              "      <th>BioSample</th>\n",
              "      <th>BioSampleModel</th>\n",
              "      <th>Center_Name</th>\n",
              "      <th>DATASTORE_filetype</th>\n",
              "      <th>DATASTORE_provider</th>\n",
              "      <th>DATASTORE_region</th>\n",
              "      <th>Experiment</th>\n",
              "      <th>Instrument</th>\n",
              "      <th>LibraryLayout</th>\n",
              "      <th>...</th>\n",
              "      <th>potential_contaminant</th>\n",
              "      <th>sequence_type</th>\n",
              "      <th>serovar</th>\n",
              "      <th>strain</th>\n",
              "      <th>Assay_Type</th>\n",
              "      <th>BioProject</th>\n",
              "      <th>Consent</th>\n",
              "      <th>InsertSize</th>\n",
              "      <th>LibrarySource</th>\n",
              "      <th>SRA_Study</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>5070</th>\n",
              "      <td>182</td>\n",
              "      <td>SAMN06824382</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755083</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352161</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5075</th>\n",
              "      <td>195</td>\n",
              "      <td>SAMN06826857</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755098</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352162</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5082</th>\n",
              "      <td>194</td>\n",
              "      <td>SAMN06826858</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755103</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>346592</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5090</th>\n",
              "      <td>186</td>\n",
              "      <td>SAMN06826859</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755111</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>278</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>347988</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5091</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN06826856</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755113</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>346590</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5092</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN06826850</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755114</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352111</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5093</th>\n",
              "      <td>189</td>\n",
              "      <td>SAMN06826854</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755115</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>23</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>345727</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5094</th>\n",
              "      <td>194</td>\n",
              "      <td>SAMN06826855</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755116</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>346594</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5095</th>\n",
              "      <td>185</td>\n",
              "      <td>SAMN06826848</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755117</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>23</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>345726</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5096</th>\n",
              "      <td>192</td>\n",
              "      <td>SAMN06826852</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755118</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7010</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>346591</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5097</th>\n",
              "      <td>195</td>\n",
              "      <td>SAMN06826853</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755119</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352145</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5098</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN06826849</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755120</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352123</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5099</th>\n",
              "      <td>181</td>\n",
              "      <td>SAMN06826851</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755121</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>278</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>347989</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5104</th>\n",
              "      <td>195</td>\n",
              "      <td>SAMN06826864</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX2755149</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>7077</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>352093</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5316</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN07672211</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198674</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>142165</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5321</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN07672204</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198679</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>149351</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5329</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN07672229</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198687</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>141358</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5333</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN07672215</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198691</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>-</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>141356</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5381</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN07672274</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198784</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>-</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>141355</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5410</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN07672304</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198813</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>282679</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5567</th>\n",
              "      <td>193</td>\n",
              "      <td>SAMN07672564</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198976</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>333400</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5574</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN07672581</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3198984</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>323043</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5671</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN07672651</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3199113</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>342959</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5713</th>\n",
              "      <td>185</td>\n",
              "      <td>SAMN07672727</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3199661</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>347351</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5773</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN07672944</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3199725</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>357999</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6042</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN07709018</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3218433</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>401708</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6075</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN07807497</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3295972</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>1727</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>388703</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6089</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN07807509</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3296002</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>Novel</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>363622</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6108</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN07807524</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX3296023</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>201</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>381675</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6131</th>\n",
              "      <td>181</td>\n",
              "      <td>SAMN08100227</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX3423888</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>not_checked</td>\n",
              "      <td>216</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>347970</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10592</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN09634124</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4370528</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>Novel</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>467324</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10593</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN09634162</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4370542</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>155</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>329906</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10608</th>\n",
              "      <td>184</td>\n",
              "      <td>SAMN09637275</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375114</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>Novel</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>347973</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10609</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN09637269</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375111</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>21</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>531329</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10610</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN09637288</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375117</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>164</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>363623</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10611</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN09637276</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375119</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>155</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>423931</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10613</th>\n",
              "      <td>179</td>\n",
              "      <td>SAMN09637285</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375125</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>6701</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>366355</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10614</th>\n",
              "      <td>197</td>\n",
              "      <td>SAMN09637289</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375120</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>6701</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>371890</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10617</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN09637282</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375126</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>408608</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10625</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN09637349</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375215</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>408644</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10630</th>\n",
              "      <td>194</td>\n",
              "      <td>SAMN09637326</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375219</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>6701</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>370895</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10633</th>\n",
              "      <td>196</td>\n",
              "      <td>SAMN09637337</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375223</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>315621</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10635</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN09637344</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375226</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>539890</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10636</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN09637309</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375225</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>Novel</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>323846</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10638</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN09637308</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375228</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>408646</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10645</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN09637312</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375235</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>323044</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10646</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN09637332</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4375236</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>6265</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>550792</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10647</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN09643596</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4381401</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>3744</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>446540</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10689</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN09666225</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4395945</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>570829</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10973</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN10065288</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4685549</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>2715</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>587807</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11319</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10075530</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4700788</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>2715</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>591878</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11618</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10106073</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4730108</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>608466</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11623</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN10106115</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4730116</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>608469</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11931</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10269645</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4913560</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>623540</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11954</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10269667</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX4913602</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>11</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>623542</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12434</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN10474769</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi sra-sos</td>\n",
              "      <td>sra-sos.public</td>\n",
              "      <td>SRX5059222</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>329</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>643141</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12519</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10521787</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5095916</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>647814</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12533</th>\n",
              "      <td>200</td>\n",
              "      <td>SAMN10535808</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5108062</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>343</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>650065</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12554</th>\n",
              "      <td>199</td>\n",
              "      <td>SAMN10535832</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5108083</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>10</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>648967</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12740</th>\n",
              "      <td>198</td>\n",
              "      <td>SAMN10723574</td>\n",
              "      <td>Pathogen.env</td>\n",
              "      <td>PHE</td>\n",
              "      <td>minhash_sig realign ref_stats sra wgmlst_sig</td>\n",
              "      <td>ncbi</td>\n",
              "      <td>NaN</td>\n",
              "      <td>SRX5237639</td>\n",
              "      <td>Illumina HiSeq 2500</td>\n",
              "      <td>PAIRED</td>\n",
              "      <td>...</td>\n",
              "      <td>None detected</td>\n",
              "      <td>43</td>\n",
              "      <td>Escherichia coli</td>\n",
              "      <td>661095</td>\n",
              "      <td>WGS</td>\n",
              "      <td>PRJNA315192</td>\n",
              "      <td>public</td>\n",
              "      <td>0</td>\n",
              "      <td>GENOMIC</td>\n",
              "      <td>SRP071789</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>113 rows × 39 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "       AvgSpotLen     BioSample BioSampleModel Center_Name  \\\n",
              "5070          182  SAMN06824382   Pathogen.env         PHE   \n",
              "5075          195  SAMN06826857   Pathogen.env         PHE   \n",
              "5082          194  SAMN06826858   Pathogen.env         PHE   \n",
              "5090          186  SAMN06826859   Pathogen.env         PHE   \n",
              "5091          197  SAMN06826856   Pathogen.env         PHE   \n",
              "5092          196  SAMN06826850   Pathogen.env         PHE   \n",
              "5093          189  SAMN06826854   Pathogen.env         PHE   \n",
              "5094          194  SAMN06826855   Pathogen.env         PHE   \n",
              "5095          185  SAMN06826848   Pathogen.env         PHE   \n",
              "5096          192  SAMN06826852   Pathogen.env         PHE   \n",
              "5097          195  SAMN06826853   Pathogen.env         PHE   \n",
              "5098          196  SAMN06826849   Pathogen.env         PHE   \n",
              "5099          181  SAMN06826851   Pathogen.env         PHE   \n",
              "5104          195  SAMN06826864   Pathogen.env         PHE   \n",
              "5316          200  SAMN07672211   Pathogen.env         PHE   \n",
              "5321          200  SAMN07672204   Pathogen.env         PHE   \n",
              "5329          200  SAMN07672229   Pathogen.env         PHE   \n",
              "5333          200  SAMN07672215   Pathogen.env         PHE   \n",
              "5381          200  SAMN07672274   Pathogen.env         PHE   \n",
              "5410          196  SAMN07672304   Pathogen.env         PHE   \n",
              "5567          193  SAMN07672564   Pathogen.env         PHE   \n",
              "5574          198  SAMN07672581   Pathogen.env         PHE   \n",
              "5671          198  SAMN07672651   Pathogen.env         PHE   \n",
              "5713          185  SAMN07672727   Pathogen.env         PHE   \n",
              "5773          199  SAMN07672944   Pathogen.env         PHE   \n",
              "6042          199  SAMN07709018   Pathogen.env         PHE   \n",
              "6075          197  SAMN07807497   Pathogen.env         PHE   \n",
              "6089          197  SAMN07807509   Pathogen.env         PHE   \n",
              "6108          196  SAMN07807524   Pathogen.env         PHE   \n",
              "6131          181  SAMN08100227   Pathogen.env         PHE   \n",
              "...           ...           ...            ...         ...   \n",
              "10592         196  SAMN09634124   Pathogen.env         PHE   \n",
              "10593         196  SAMN09634162   Pathogen.env         PHE   \n",
              "10608         184  SAMN09637275   Pathogen.env         PHE   \n",
              "10609         200  SAMN09637269   Pathogen.env         PHE   \n",
              "10610         197  SAMN09637288   Pathogen.env         PHE   \n",
              "10611         199  SAMN09637276   Pathogen.env         PHE   \n",
              "10613         179  SAMN09637285   Pathogen.env         PHE   \n",
              "10614         197  SAMN09637289   Pathogen.env         PHE   \n",
              "10617         200  SAMN09637282   Pathogen.env         PHE   \n",
              "10625         199  SAMN09637349   Pathogen.env         PHE   \n",
              "10630         194  SAMN09637326   Pathogen.env         PHE   \n",
              "10633         196  SAMN09637337   Pathogen.env         PHE   \n",
              "10635         200  SAMN09637344   Pathogen.env         PHE   \n",
              "10636         198  SAMN09637309   Pathogen.env         PHE   \n",
              "10638         200  SAMN09637308   Pathogen.env         PHE   \n",
              "10645         198  SAMN09637312   Pathogen.env         PHE   \n",
              "10646         200  SAMN09637332   Pathogen.env         PHE   \n",
              "10647         199  SAMN09643596   Pathogen.env         PHE   \n",
              "10689         199  SAMN09666225   Pathogen.env         PHE   \n",
              "10973         199  SAMN10065288   Pathogen.env         PHE   \n",
              "11319         200  SAMN10075530   Pathogen.env         PHE   \n",
              "11618         200  SAMN10106073   Pathogen.env         PHE   \n",
              "11623         199  SAMN10106115   Pathogen.env         PHE   \n",
              "11931         200  SAMN10269645   Pathogen.env         PHE   \n",
              "11954         200  SAMN10269667   Pathogen.env         PHE   \n",
              "12434         199  SAMN10474769   Pathogen.env         PHE   \n",
              "12519         200  SAMN10521787   Pathogen.env         PHE   \n",
              "12533         200  SAMN10535808   Pathogen.env         PHE   \n",
              "12554         199  SAMN10535832   Pathogen.env         PHE   \n",
              "12740         198  SAMN10723574   Pathogen.env         PHE   \n",
              "\n",
              "                                 DATASTORE_filetype DATASTORE_provider  \\\n",
              "5070   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5075   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5082   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5090   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5091   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5092   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5093   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5094   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5095   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5096   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5097   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5098   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5099   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5104   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5316   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5321   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5329   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5333   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5381   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5410   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5567   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5574   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5671   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5713   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "5773   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "6042   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "6075   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "6089   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "6108   minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "6131   minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "...                                             ...                ...   \n",
              "10592  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10593  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10608  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10609  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10610  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10611  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10613  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10614  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10617  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10625  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10630  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10633  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10635  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10636  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10638  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10645  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10646  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10647  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10689  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "10973  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "11319  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "11618  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "11623  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "11931  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "11954  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "12434  minhash_sig realign ref_stats sra wgmlst_sig       ncbi sra-sos   \n",
              "12519  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "12533  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "12554  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "12740  minhash_sig realign ref_stats sra wgmlst_sig               ncbi   \n",
              "\n",
              "      DATASTORE_region  Experiment           Instrument LibraryLayout  \\\n",
              "5070               NaN  SRX2755083  Illumina HiSeq 2500        PAIRED   \n",
              "5075               NaN  SRX2755098  Illumina HiSeq 2500        PAIRED   \n",
              "5082               NaN  SRX2755103  Illumina HiSeq 2500        PAIRED   \n",
              "5090               NaN  SRX2755111  Illumina HiSeq 2500        PAIRED   \n",
              "5091               NaN  SRX2755113  Illumina HiSeq 2500        PAIRED   \n",
              "5092               NaN  SRX2755114  Illumina HiSeq 2500        PAIRED   \n",
              "5093               NaN  SRX2755115  Illumina HiSeq 2500        PAIRED   \n",
              "5094               NaN  SRX2755116  Illumina HiSeq 2500        PAIRED   \n",
              "5095               NaN  SRX2755117  Illumina HiSeq 2500        PAIRED   \n",
              "5096               NaN  SRX2755118  Illumina HiSeq 2500        PAIRED   \n",
              "5097               NaN  SRX2755119  Illumina HiSeq 2500        PAIRED   \n",
              "5098               NaN  SRX2755120  Illumina HiSeq 2500        PAIRED   \n",
              "5099               NaN  SRX2755121  Illumina HiSeq 2500        PAIRED   \n",
              "5104               NaN  SRX2755149  Illumina HiSeq 2500        PAIRED   \n",
              "5316               NaN  SRX3198674  Illumina HiSeq 2500        PAIRED   \n",
              "5321               NaN  SRX3198679  Illumina HiSeq 2500        PAIRED   \n",
              "5329               NaN  SRX3198687  Illumina HiSeq 2500        PAIRED   \n",
              "5333               NaN  SRX3198691  Illumina HiSeq 2500        PAIRED   \n",
              "5381               NaN  SRX3198784  Illumina HiSeq 2500        PAIRED   \n",
              "5410               NaN  SRX3198813  Illumina HiSeq 2500        PAIRED   \n",
              "5567               NaN  SRX3198976  Illumina HiSeq 2500        PAIRED   \n",
              "5574               NaN  SRX3198984  Illumina HiSeq 2500        PAIRED   \n",
              "5671               NaN  SRX3199113  Illumina HiSeq 2500        PAIRED   \n",
              "5713               NaN  SRX3199661  Illumina HiSeq 2500        PAIRED   \n",
              "5773               NaN  SRX3199725  Illumina HiSeq 2500        PAIRED   \n",
              "6042               NaN  SRX3218433  Illumina HiSeq 2500        PAIRED   \n",
              "6075               NaN  SRX3295972  Illumina HiSeq 2500        PAIRED   \n",
              "6089               NaN  SRX3296002  Illumina HiSeq 2500        PAIRED   \n",
              "6108               NaN  SRX3296023  Illumina HiSeq 2500        PAIRED   \n",
              "6131    sra-sos.public  SRX3423888  Illumina HiSeq 2500        PAIRED   \n",
              "...                ...         ...                  ...           ...   \n",
              "10592              NaN  SRX4370528  Illumina HiSeq 2500        PAIRED   \n",
              "10593              NaN  SRX4370542  Illumina HiSeq 2500        PAIRED   \n",
              "10608              NaN  SRX4375114  Illumina HiSeq 2500        PAIRED   \n",
              "10609              NaN  SRX4375111  Illumina HiSeq 2500        PAIRED   \n",
              "10610              NaN  SRX4375117  Illumina HiSeq 2500        PAIRED   \n",
              "10611              NaN  SRX4375119  Illumina HiSeq 2500        PAIRED   \n",
              "10613              NaN  SRX4375125  Illumina HiSeq 2500        PAIRED   \n",
              "10614              NaN  SRX4375120  Illumina HiSeq 2500        PAIRED   \n",
              "10617              NaN  SRX4375126  Illumina HiSeq 2500        PAIRED   \n",
              "10625              NaN  SRX4375215  Illumina HiSeq 2500        PAIRED   \n",
              "10630              NaN  SRX4375219  Illumina HiSeq 2500        PAIRED   \n",
              "10633              NaN  SRX4375223  Illumina HiSeq 2500        PAIRED   \n",
              "10635              NaN  SRX4375226  Illumina HiSeq 2500        PAIRED   \n",
              "10636              NaN  SRX4375225  Illumina HiSeq 2500        PAIRED   \n",
              "10638              NaN  SRX4375228  Illumina HiSeq 2500        PAIRED   \n",
              "10645              NaN  SRX4375235  Illumina HiSeq 2500        PAIRED   \n",
              "10646              NaN  SRX4375236  Illumina HiSeq 2500        PAIRED   \n",
              "10647              NaN  SRX4381401  Illumina HiSeq 2500        PAIRED   \n",
              "10689              NaN  SRX4395945  Illumina HiSeq 2500        PAIRED   \n",
              "10973              NaN  SRX4685549  Illumina HiSeq 2500        PAIRED   \n",
              "11319              NaN  SRX4700788  Illumina HiSeq 2500        PAIRED   \n",
              "11618              NaN  SRX4730108  Illumina HiSeq 2500        PAIRED   \n",
              "11623              NaN  SRX4730116  Illumina HiSeq 2500        PAIRED   \n",
              "11931              NaN  SRX4913560  Illumina HiSeq 2500        PAIRED   \n",
              "11954              NaN  SRX4913602  Illumina HiSeq 2500        PAIRED   \n",
              "12434   sra-sos.public  SRX5059222  Illumina HiSeq 2500        PAIRED   \n",
              "12519              NaN  SRX5095916  Illumina HiSeq 2500        PAIRED   \n",
              "12533              NaN  SRX5108062  Illumina HiSeq 2500        PAIRED   \n",
              "12554              NaN  SRX5108083  Illumina HiSeq 2500        PAIRED   \n",
              "12740              NaN  SRX5237639  Illumina HiSeq 2500        PAIRED   \n",
              "\n",
              "         ...     potential_contaminant sequence_type           serovar  \\\n",
              "5070     ...               not_checked          7077  Escherichia coli   \n",
              "5075     ...               not_checked          7077  Escherichia coli   \n",
              "5082     ...               not_checked          7077  Escherichia coli   \n",
              "5090     ...               not_checked           278  Escherichia coli   \n",
              "5091     ...               not_checked          7077  Escherichia coli   \n",
              "5092     ...               not_checked          7077  Escherichia coli   \n",
              "5093     ...               not_checked            23  Escherichia coli   \n",
              "5094     ...               not_checked          7077  Escherichia coli   \n",
              "5095     ...               not_checked            23  Escherichia coli   \n",
              "5096     ...               not_checked          7010  Escherichia coli   \n",
              "5097     ...               not_checked          7077  Escherichia coli   \n",
              "5098     ...               not_checked          7077  Escherichia coli   \n",
              "5099     ...               not_checked           278  Escherichia coli   \n",
              "5104     ...               not_checked          7077  Escherichia coli   \n",
              "5316     ...               not_checked            11  Escherichia coli   \n",
              "5321     ...               not_checked            11  Escherichia coli   \n",
              "5329     ...               not_checked            11  Escherichia coli   \n",
              "5333     ...               not_checked             -  Escherichia coli   \n",
              "5381     ...               not_checked             -  Escherichia coli   \n",
              "5410     ...               not_checked            11  Escherichia coli   \n",
              "5567     ...               not_checked            11  Escherichia coli   \n",
              "5574     ...               not_checked            11  Escherichia coli   \n",
              "5671     ...               not_checked            11  Escherichia coli   \n",
              "5713     ...               not_checked            11  Escherichia coli   \n",
              "5773     ...               not_checked            11  Escherichia coli   \n",
              "6042     ...               not_checked            11  Escherichia coli   \n",
              "6075     ...               not_checked          1727  Escherichia coli   \n",
              "6089     ...               not_checked         Novel  Escherichia coli   \n",
              "6108     ...               not_checked           201  Escherichia coli   \n",
              "6131     ...               not_checked           216  Escherichia coli   \n",
              "...      ...                       ...           ...               ...   \n",
              "10592    ...             None detected         Novel  Escherichia coli   \n",
              "10593    ...             None detected           155  Escherichia coli   \n",
              "10608    ...             None detected         Novel  Escherichia coli   \n",
              "10609    ...             None detected            21  Escherichia coli   \n",
              "10610    ...             None detected           164  Escherichia coli   \n",
              "10611    ...             None detected           155  Escherichia coli   \n",
              "10613    ...             None detected          6701  Escherichia coli   \n",
              "10614    ...             None detected          6701  Escherichia coli   \n",
              "10617    ...             None detected            10  Escherichia coli   \n",
              "10625    ...             None detected            10  Escherichia coli   \n",
              "10630    ...             None detected          6701  Escherichia coli   \n",
              "10633    ...             None detected            10  Escherichia coli   \n",
              "10635    ...             None detected            11  Escherichia coli   \n",
              "10636    ...             None detected         Novel  Escherichia coli   \n",
              "10638    ...             None detected            10  Escherichia coli   \n",
              "10645    ...             None detected            11  Escherichia coli   \n",
              "10646    ...             None detected          6265  Escherichia coli   \n",
              "10647    ...             None detected          3744  Escherichia coli   \n",
              "10689    ...             None detected            10  Escherichia coli   \n",
              "10973    ...             None detected          2715  Escherichia coli   \n",
              "11319    ...             None detected          2715  Escherichia coli   \n",
              "11618    ...             None detected            11  Escherichia coli   \n",
              "11623    ...             None detected            11  Escherichia coli   \n",
              "11931    ...             None detected            11  Escherichia coli   \n",
              "11954    ...             None detected            11  Escherichia coli   \n",
              "12434    ...             None detected           329  Escherichia coli   \n",
              "12519    ...             None detected            10  Escherichia coli   \n",
              "12533    ...             None detected           343  Escherichia coli   \n",
              "12554    ...             None detected            10  Escherichia coli   \n",
              "12740    ...             None detected            43  Escherichia coli   \n",
              "\n",
              "       strain  Assay_Type   BioProject Consent InsertSize LibrarySource  \\\n",
              "5070   352161         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5075   352162         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5082   346592         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5090   347988         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5091   346590         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5092   352111         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5093   345727         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5094   346594         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5095   345726         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5096   346591         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5097   352145         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5098   352123         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5099   347989         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5104   352093         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5316   142165         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5321   149351         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5329   141358         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5333   141356         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5381   141355         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5410   282679         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5567   333400         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5574   323043         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5671   342959         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5713   347351         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "5773   357999         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6042   401708         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6075   388703         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6089   363622         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6108   381675         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "6131   347970         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "...       ...         ...          ...     ...        ...           ...   \n",
              "10592  467324         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10593  329906         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10608  347973         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10609  531329         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10610  363623         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10611  423931         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10613  366355         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10614  371890         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10617  408608         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10625  408644         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10630  370895         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10633  315621         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10635  539890         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10636  323846         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10638  408646         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10645  323044         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10646  550792         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10647  446540         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10689  570829         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "10973  587807         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11319  591878         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11618  608466         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11623  608469         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11931  623540         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "11954  623542         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12434  643141         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12519  647814         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12533  650065         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12554  648967         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "12740  661095         WGS  PRJNA315192  public          0       GENOMIC   \n",
              "\n",
              "       SRA_Study  \n",
              "5070   SRP071789  \n",
              "5075   SRP071789  \n",
              "5082   SRP071789  \n",
              "5090   SRP071789  \n",
              "5091   SRP071789  \n",
              "5092   SRP071789  \n",
              "5093   SRP071789  \n",
              "5094   SRP071789  \n",
              "5095   SRP071789  \n",
              "5096   SRP071789  \n",
              "5097   SRP071789  \n",
              "5098   SRP071789  \n",
              "5099   SRP071789  \n",
              "5104   SRP071789  \n",
              "5316   SRP071789  \n",
              "5321   SRP071789  \n",
              "5329   SRP071789  \n",
              "5333   SRP071789  \n",
              "5381   SRP071789  \n",
              "5410   SRP071789  \n",
              "5567   SRP071789  \n",
              "5574   SRP071789  \n",
              "5671   SRP071789  \n",
              "5713   SRP071789  \n",
              "5773   SRP071789  \n",
              "6042   SRP071789  \n",
              "6075   SRP071789  \n",
              "6089   SRP071789  \n",
              "6108   SRP071789  \n",
              "6131   SRP071789  \n",
              "...          ...  \n",
              "10592  SRP071789  \n",
              "10593  SRP071789  \n",
              "10608  SRP071789  \n",
              "10609  SRP071789  \n",
              "10610  SRP071789  \n",
              "10611  SRP071789  \n",
              "10613  SRP071789  \n",
              "10614  SRP071789  \n",
              "10617  SRP071789  \n",
              "10625  SRP071789  \n",
              "10630  SRP071789  \n",
              "10633  SRP071789  \n",
              "10635  SRP071789  \n",
              "10636  SRP071789  \n",
              "10638  SRP071789  \n",
              "10645  SRP071789  \n",
              "10646  SRP071789  \n",
              "10647  SRP071789  \n",
              "10689  SRP071789  \n",
              "10973  SRP071789  \n",
              "11319  SRP071789  \n",
              "11618  SRP071789  \n",
              "11623  SRP071789  \n",
              "11931  SRP071789  \n",
              "11954  SRP071789  \n",
              "12434  SRP071789  \n",
              "12519  SRP071789  \n",
              "12533  SRP071789  \n",
              "12554  SRP071789  \n",
              "12740  SRP071789  \n",
              "\n",
              "[113 rows x 39 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 24
        }
      ]
    },
    {
      "metadata": {
        "id": "Rxeuyludkan9",
        "colab_type": "code",
        "outputId": "2fd1be0c-8c42-47ff-95fe-767725e0c0ec",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1&m2]['isolation_source'].unique()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['food'], dtype=object)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 25
        }
      ]
    },
    {
      "metadata": {
        "id": "RP1K02PIk160",
        "colab_type": "code",
        "outputId": "27381ddd-7078-4f78-9667-afc8e99fa1c6",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra.shape"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "(13175, 39)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 26
        }
      ]
    },
    {
      "metadata": {
        "id": "lFKfor1Pk5CO",
        "colab_type": "code",
        "outputId": "deec2ff6-a759-46a1-f323-b6610cb1e446",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra[m1&m2].shape\n",
        "# data의 양이 줄음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "(113, 39)"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 27
        }
      ]
    },
    {
      "metadata": {
        "id": "Kghugv2KlNkj",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "df_sra_ecoli_food = df_sra[m1&m2]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "ECfCOojVlVLU",
        "colab_type": "code",
        "outputId": "8ff683f5-94b8-44c1-af57-bdd6b367eca8",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1108
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra_ecoli_food['Run']"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "5070     SRR5470223\n",
              "5075     SRR5470238\n",
              "5082     SRR5470243\n",
              "5090     SRR5470251\n",
              "5091     SRR5470253\n",
              "5092     SRR5470254\n",
              "5093     SRR5470255\n",
              "5094     SRR5470256\n",
              "5095     SRR5470257\n",
              "5096     SRR5470258\n",
              "5097     SRR5470259\n",
              "5098     SRR5470260\n",
              "5099     SRR5470261\n",
              "5104     SRR5470345\n",
              "5316     SRR6051734\n",
              "5321     SRR6051739\n",
              "5329     SRR6051747\n",
              "5333     SRR6051751\n",
              "5381     SRR6051844\n",
              "5410     SRR6051873\n",
              "5567     SRR6052036\n",
              "5574     SRR6052044\n",
              "5671     SRR6052173\n",
              "5713     SRR6052721\n",
              "5773     SRR6052785\n",
              "6042     SRR6079445\n",
              "6075     SRR6185826\n",
              "6089     SRR6185855\n",
              "6108     SRR6185875\n",
              "6131     SRR6324135\n",
              "            ...    \n",
              "10592    SRR7501167\n",
              "10593    SRR7501181\n",
              "10608    SRR7505768\n",
              "10609    SRR7505765\n",
              "10610    SRR7505771\n",
              "10611    SRR7505773\n",
              "10613    SRR7505779\n",
              "10614    SRR7505774\n",
              "10617    SRR7505780\n",
              "10625    SRR7505869\n",
              "10630    SRR7505873\n",
              "10633    SRR7505877\n",
              "10635    SRR7505880\n",
              "10636    SRR7505879\n",
              "10638    SRR7505882\n",
              "10645    SRR7505889\n",
              "10646    SRR7505890\n",
              "10647    SRR7511299\n",
              "10689    SRR7527615\n",
              "10973    SRR7834586\n",
              "11319    SRR7850029\n",
              "11618    SRR7892531\n",
              "11623    SRR7892539\n",
              "11931    SRR8086651\n",
              "11954    SRR8086696\n",
              "12434    SRR8240970\n",
              "12519    SRR8281111\n",
              "12533    SRR8293474\n",
              "12554    SRR8293495\n",
              "12740    SRR8429822\n",
              "Name: Run, Length: 113, dtype: object"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 29
        }
      ]
    },
    {
      "metadata": {
        "id": "IRPa7_2qlYUz",
        "colab_type": "code",
        "outputId": "2429424f-55f8-4322-88c9-4f1e223c2eac",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        }
      },
      "cell_type": "code",
      "source": [
        "df_sra_ecoli_food['Run'].head()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "5070    SRR5470223\n",
              "5075    SRR5470238\n",
              "5082    SRR5470243\n",
              "5090    SRR5470251\n",
              "5091    SRR5470253\n",
              "Name: Run, dtype: object"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 30
        }
      ]
    },
    {
      "metadata": {
        "id": "3Ly0GLgrlkSQ",
        "colab_type": "text"
      },
      "cell_type": "markdown",
      "source": [
        "# Download one SRR file from NCBI and make BAM file"
      ]
    },
    {
      "metadata": {
        "id": "ixIZhJebmCyu",
        "colab_type": "code",
        "outputId": "92807ae6-41e3-4edd-abdc-232852fe1a72",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 318
        }
      },
      "cell_type": "code",
      "source": [
        "# Downloading a SRR file\n",
        "# SRR ID를 변수에 지정하면 VCF를 만드는 명령어들 속에 SRR ID를 직접 입력하지 않고 변수를 사용하면 되기 때문에 좀 더 편리하다\n",
        "query_sra = 'SRR5470223'\n",
        "sra_dir1 = query_sra[0:3]\n",
        "sra_dir2 = query_sra[0:6]\n",
        "sra_dir3 = query_sra \n",
        "sra_file = query_sra+'.sra'\n",
        "print('Downloading',query_sra)\n",
        "!wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/{sra_dir1}/{sra_dir2}/{sra_dir3}/{sra_file}"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Downloading SRR5470223\n",
            "--2019-03-26 06:06:18--  ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR547/SRR5470223/SRR5470223.sra\n",
            "           => ‘SRR5470223.sra’\n",
            "Resolving ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)... 130.14.250.11, 2607:f220:41e:250::7\n",
            "Connecting to ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)|130.14.250.11|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /sra/sra-instant/reads/ByRun/sra/SRR/SRR547/SRR5470223 ... done.\n",
            "==> SIZE SRR5470223.sra ... 60074833\n",
            "==> PASV ... done.    ==> RETR SRR5470223.sra ... done.\n",
            "Length: 60074833 (57M) (unauthoritative)\n",
            "\n",
            "SRR5470223.sra      100%[===================>]  57.29M  20.6MB/s    in 2.8s    \n",
            "\n",
            "2019-03-26 06:06:21 (20.6 MB/s) - ‘SRR5470223.sra’ saved [60074833]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "Aa6q9BdGmP9K",
        "colab_type": "text"
      },
      "cell_type": "markdown",
      "source": [
        "---\n",
        "sra_id = 'SRR5470223'\n",
        "\n",
        "! wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/{sra_id[0:3]}/{sra_id[0:6]}/{sra_id}/{sra_id}.sra\n",
        "\n",
        "---"
      ]
    },
    {
      "metadata": {
        "id": "gRdRyojtmT6L",
        "colab_type": "code",
        "outputId": "f10ecd98-5ba2-4226-b336-b0191204a3c8",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        }
      },
      "cell_type": "code",
      "source": [
        "ls"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "SRP071789.txt  SRR5470223.sra\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "vH0hoiMznF-I",
        "colab_type": "code",
        "outputId": "19791a52-d960-4519-f88f-13966e447889",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 3909
        }
      },
      "cell_type": "code",
      "source": [
        "# Converting SRR to fastq\n",
        "! fastq-dump  --origfmt -I  --split-files --gzip {query_sra}.sra\n",
        "# Mapping fastq to reference\n",
        "! bwa mem ../references/GCA_004328685.1_ASM432868v1_genomic.fna {query_sra}_1.fastq.gz {query_sra}_2.fastq.gz -o {query_sra}.sam\n",
        "# paired end mapping correction \n",
        "! samtools fixmate -O bam {query_sra}.sam {query_sra}.bam\n",
        "# sort the alignment file\n",
        "! samtools sort -o {query_sra}.sorted.bam  {query_sra}.bam"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Read 749410 spots for SRR5470223.sra\n",
            "Written 749410 spots for SRR5470223.sra\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 109560 sequences (10000066 bp)...\n",
            "[M::process] read 109736 sequences (10000058 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 47625, 3, 8)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 123, 170)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 336)\n",
            "[M::mem_pestat] mean and std.dev: (131.91, 61.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 419)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 109560 reads in 9.051 CPU sec, 8.841 real sec\n",
            "[M::process] read 109734 sequences (10000059 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 47732, 6, 9)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 123, 170)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 336)\n",
            "[M::mem_pestat] mean and std.dev: (131.68, 61.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 419)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 109736 reads in 8.812 CPU sec, 8.415 real sec\n",
            "[M::process] read 108982 sequences (10000111 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 47650, 7, 11)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (52, 76, 141)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
            "[M::mem_pestat] mean and std.dev: (83.62, 38.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (86, 122, 169)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 335)\n",
            "[M::mem_pestat] mean and std.dev: (131.25, 61.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 418)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (77, 87, 564)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1538)\n",
            "[M::mem_pestat] mean and std.dev: (142.56, 155.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2025)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 109734 reads in 8.471 CPU sec, 8.122 real sec\n",
            "[M::process] read 109594 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 47272, 5, 10)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (89, 124, 173)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 341)\n",
            "[M::mem_pestat] mean and std.dev: (134.07, 62.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 425)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (80, 167, 500)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1340)\n",
            "[M::mem_pestat] mean and std.dev: (204.11, 173.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1760)\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 108982 reads in 8.272 CPU sec, 7.929 real sec\n",
            "[M::process] read 109796 sequences (10000041 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 47647, 6, 12)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (78, 118, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
            "[M::mem_pestat] mean and std.dev: (104.00, 50.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 678)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 124, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 339)\n",
            "[M::mem_pestat] mean and std.dev: (132.78, 62.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 423)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (80, 166, 571)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1553)\n",
            "[M::mem_pestat] mean and std.dev: (189.30, 177.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2044)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 109594 reads in 8.275 CPU sec, 7.916 real sec\n",
            "[M::process] read 109170 sequences (10000001 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 47730, 7, 10)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (49, 69, 90)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 172)\n",
            "[M::mem_pestat] mean and std.dev: (62.60, 22.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 213)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (86, 122, 170)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 338)\n",
            "[M::mem_pestat] mean and std.dev: (131.60, 61.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 422)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (68, 102, 173)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 383)\n",
            "[M::mem_pestat] mean and std.dev: (117.90, 56.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 488)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 109796 reads in 8.426 CPU sec, 8.146 real sec\n",
            "[M::process] read 109752 sequences (10000077 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 47222, 6, 11)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (88, 124, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 337)\n",
            "[M::mem_pestat] mean and std.dev: (133.18, 61.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 420)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (75, 102, 169)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 357)\n",
            "[M::mem_pestat] mean and std.dev: (89.78, 41.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 451)\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 109170 reads in 8.368 CPU sec, 8.071 real sec\n",
            "[M::process] read 109586 sequences (10000034 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 47599, 6, 9)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 123, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 339)\n",
            "[M::mem_pestat] mean and std.dev: (131.85, 61.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 423)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 109752 reads in 8.294 CPU sec, 8.037 real sec\n",
            "[M::process] read 109658 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 47602, 7, 5)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (62, 72, 98)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 170)\n",
            "[M::mem_pestat] mean and std.dev: (73.30, 26.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 206)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 123, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 339)\n",
            "[M::mem_pestat] mean and std.dev: (132.17, 61.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 423)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_process_seqs] Processed 109586 reads in 8.614 CPU sec, 8.303 real sec\n",
            "[M::process] read 109176 sequences (10000170 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 47634, 4, 8)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 122, 170)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 336)\n",
            "[M::mem_pestat] mean and std.dev: (131.70, 61.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 419)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 109658 reads in 8.611 CPU sec, 8.325 real sec\n",
            "[M::process] read 109462 sequences (10000145 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 47332, 3, 7)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (53, 94, 133)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 293)\n",
            "[M::mem_pestat] mean and std.dev: (82.27, 36.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 373)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (88, 124, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 337)\n",
            "[M::mem_pestat] mean and std.dev: (132.66, 61.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 420)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_process_seqs] Processed 109176 reads in 8.435 CPU sec, 8.139 real sec\n",
            "[M::process] read 109690 sequences (10000021 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 47456, 6, 11)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (70, 92, 305)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 775)\n",
            "[M::mem_pestat] mean and std.dev: (136.92, 115.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1010)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 124, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 339)\n",
            "[M::mem_pestat] mean and std.dev: (132.77, 61.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 423)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 126, 171)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 339)\n",
            "[M::mem_pestat] mean and std.dev: (122.60, 74.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 423)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 109462 reads in 8.411 CPU sec, 8.134 real sec\n",
            "[M::process] read 74924 sequences (6862294 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 47682, 7, 6)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (87, 123, 170)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 336)\n",
            "[M::mem_pestat] mean and std.dev: (131.76, 61.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 419)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 109690 reads in 8.298 CPU sec, 8.053 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 32502, 4, 3)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (88, 124, 172)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 340)\n",
            "[M::mem_pestat] mean and std.dev: (133.30, 62.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 424)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 74924 reads in 5.655 CPU sec, 5.587 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem -o SRR5470223.sam ../references/GCA_004328685.1_ASM432868v1_genomic.fna SRR5470223_1.fastq.gz SRR5470223_2.fastq.gz\n",
            "[main] Real time: 114.160 sec; CPU: 116.343 sec\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "gfMCwW9WnkJv",
        "colab_type": "text"
      },
      "cell_type": "markdown",
      "source": [
        "---\n",
        "**fastq-dump 실행파일 지정, error 방지** <br />\n",
        "! chmod +x ../sratoolkit.2.9.4-1-ubuntu64/bin/fastq-dump.2.9.4 <br />\n",
        "**Converting SRR to fastq** <br />\n",
        "! ../sratoolkit.2.9.4-1-ubuntu64/bin/fastq-dump.2.9.4 --origfmt -I --split-files --gzip ./SRR5470223.sra <br />\n",
        "**Mapping fastq to reference** <br />\n",
        "! bwa mem -o ./SRR5470223.sam ../references/GCA_004328685.1_ASM432868v1_genomic.fna ./SRR5470223_1.fastq.gz ./SRR5470223_2.fastq.gz <br />\n",
        "**paired end mapping correction** <br />\n",
        "! samtools fixmate -O bam ./SRR5470223.sam ./SRR5470223.bam <br />\n",
        "**sort the alignment file** <br />\n",
        "! samtools sort -o ./SRR5470223.sorted.bam ./SRR5470223.bam <br />\n",
        "\n",
        "---"
      ]
    },
    {
      "metadata": {
        "id": "tkKx7bahncPG",
        "colab_type": "code",
        "outputId": "c8d7e7d0-424a-4116-93d0-13ad901c2371",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 72
        }
      },
      "cell_type": "code",
      "source": [
        "# Call variations (SNP/INDEL) from BAM file\n",
        "! bcftools mpileup -Oz -f ../references/GCA_004328685.1_ASM432868v1_genomic.fna -o {query_sra}.g.vcf.gz {query_sra}.sorted.bam\n",
        "! bcftools call -mv -Ov -o {query_sra}.vcf.gz.mv.vcf {query_sra}.g.vcf.gz"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[mpileup] 1 samples in 1 input files\n",
            "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "gMiXOJPzqP_9",
        "colab_type": "text"
      },
      "cell_type": "markdown",
      "source": [
        "---\n",
        "**Call variations (SNP/INDEL) from BAM file** <br />\n",
        "! bcftools mpileup -Ov -g 10 -f ../references/GCA_004328685.1_ASM432868v1_genomic.fna -o ./SRR5470223.g.vcf ./SRR5470223.sorted.bam <br />\n",
        "! bcftools call -mv -Ov -o SRR5470223.vcf.mv.vcf SRR5470223.g.vcf <br />\n",
        "\n",
        "---"
      ]
    },
    {
      "metadata": {
        "id": "rqBTjW6wqxyo",
        "colab_type": "code",
        "outputId": "3b473bc0-10d2-46bc-a71c-25decaa79e0f",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 193
        }
      },
      "cell_type": "code",
      "source": [
        "ll\n",
        "# 만들어진 file과 용량을 보니 정상적으로 잘 만들어졌음을 볼 수 있음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "total 840065\n",
            "-rw------- 1 root   5421384 Mar 26 05:48 SRP071789.txt\n",
            "-rw------- 1 root  39961315 Mar 26 06:09 SRR5470223_1.fastq.gz\n",
            "-rw------- 1 root  40649125 Mar 26 06:09 SRR5470223_2.fastq.gz\n",
            "-rw------- 1 root 105470192 Mar 26 06:12 SRR5470223.bam\n",
            "-rw------- 1 root 104969621 Mar 26 06:22 SRR5470223.g.vcf.gz\n",
            "-rw------- 1 root 411478272 Mar 26 06:11 SRR5470223.sam\n",
            "-rw------- 1 root  76889664 Mar 26 06:12 SRR5470223.sorted.bam\n",
            "-rw------- 1 root  60074833 Mar 26 06:06 SRR5470223.sra\n",
            "-rw------- 1 root  15310265 Mar 26 06:22 SRR5470223.vcf.gz.mv.vcf\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "RtGjd43xrBPF",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "df_snp = pd.read_csv('./SRR5470223.vcf.gz.mv.vcf',sep='\\t',comment='#',header=None)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "oHUKUu-Kredc",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "columns = !cat ./SRR5470223.vcf.gz.mv.vcf | grep '#' | tail -1\n",
        "columns = columns[0].split('\\t')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "IW8UIU-grgd-",
        "colab_type": "code",
        "outputId": "a1543726-83c7-460d-87df-b7309e52a10e",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 618
        }
      },
      "cell_type": "code",
      "source": [
        "! cat ./SRR5470223.vcf.gz.mv.vcf | grep '#'"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "##fileformat=VCFv4.2\n",
            "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
            "##bcftoolsVersion=1.7+htslib-1.7-2\n",
            "##bcftoolsCommand=mpileup -Oz -f ../references/GCA_004328685.1_ASM432868v1_genomic.fna -o SRR5470223.g.vcf.gz SRR5470223.sorted.bam\n",
            "##reference=file://../references/GCA_004328685.1_ASM432868v1_genomic.fna\n",
            "##contig=<ID=CP035312.1,length=4593539>\n",
            "##contig=<ID=CP035314.1,length=89963>\n",
            "##contig=<ID=CP035316.1,length=30581>\n",
            "##contig=<ID=CP035315.1,length=40654>\n",
            "##contig=<ID=CP035313.1,length=96320>\n",
            "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
            "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
            "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of reads supporting an indel\">\n",
            "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of reads supporting an indel\">\n",
            "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Raw read depth\">\n",
            "##INFO=<ID=VDB,Number=1,Type=Float,Description=\"Variant Distance Bias for filtering splice-site artefacts in RNA-seq data (bigger is better)\",Version=\"3\">\n",
            "##INFO=<ID=RPB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Read Position Bias (bigger is better)\">\n",
            "##INFO=<ID=MQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=BQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Base Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=MQSB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality vs Strand Bias (bigger is better)\">\n",
            "##INFO=<ID=SGB,Number=1,Type=Float,Description=\"Segregation based metric.\">\n",
            "##INFO=<ID=MQ0F,Number=1,Type=Float,Description=\"Fraction of MQ0 reads (smaller is better)\">\n",
            "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"List of Phred-scaled genotype likelihoods\">\n",
            "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
            "##INFO=<ID=ICB,Number=1,Type=Float,Description=\"Inbreeding Coefficient Binomial test (bigger is better)\">\n",
            "##INFO=<ID=HOB,Number=1,Type=Float,Description=\"Bias in the number of HOMs number (smaller is better)\">\n",
            "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes for each ALT allele, in the same order as listed\">\n",
            "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
            "##INFO=<ID=DP4,Number=4,Type=Integer,Description=\"Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases\">\n",
            "##INFO=<ID=MQ,Number=1,Type=Integer,Description=\"Average mapping quality\">\n",
            "##bcftools_callVersion=1.7+htslib-1.7-2\n",
            "##bcftools_callCommand=call -mv -Ov -o SRR5470223.vcf.gz.mv.vcf SRR5470223.g.vcf.gz; Date=Tue Mar 26 06:22:41 2019\n",
            "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tSRR5470223.sorted.bam\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "ZWSzGaekriwP",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "df_snp.columns = columns"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "Kv6pBEyVrk0l",
        "colab_type": "code",
        "outputId": "d0664a64-1e0f-4eff-b736-b63479c761fd",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 215
        }
      },
      "cell_type": "code",
      "source": [
        "df_snp.head()\n",
        "\n",
        "# #CHROM : chromosome id assign, 무슨 chromosome이 있는지 알아보려면 df_snp['#CHROM'].unique() 하면 됨\n",
        "# QUAL, INFO : quality가 낮을 수록 depth(DP)의 수도 낮음\n",
        "# Read depth 26개, VDB(variant distance bias)가 0.42면 약 10개는 다르게 나온 것으로, 이 서열들이 확실하지 않다는 것\n",
        "## 16개는 G(REF)이면 10개는 T(ALT)이다. 이를 0/1로 표시, reference/variant, hetero 임을 알려줌\n",
        "# 1/1 = variant/variant homo"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>#CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>ID</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>QUAL</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>INFO</th>\n",
              "      <th>FORMAT</th>\n",
              "      <th>SRR5470223.sorted.bam</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>CP035312.1</td>\n",
              "      <td>90</td>\n",
              "      <td>.</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>3.82562</td>\n",
              "      <td>.</td>\n",
              "      <td>DP=26;VDB=0.42;SGB=-0.453602;RPB=0.235294;MQB=...</td>\n",
              "      <td>GT:PL</td>\n",
              "      <td>0/1:38,0,255</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>CP035312.1</td>\n",
              "      <td>141</td>\n",
              "      <td>.</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>30.71780</td>\n",
              "      <td>.</td>\n",
              "      <td>DP=47;VDB=0.292464;SGB=-0.511536;RPB=0.605083;...</td>\n",
              "      <td>GT:PL</td>\n",
              "      <td>0/1:67,0,255</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>CP035312.1</td>\n",
              "      <td>215</td>\n",
              "      <td>.</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>3.93785</td>\n",
              "      <td>.</td>\n",
              "      <td>DP=44;VDB=0.559209;SGB=-0.511536;RPB=0.958563;...</td>\n",
              "      <td>GT:PL</td>\n",
              "      <td>0/1:39,0,255</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>CP035312.1</td>\n",
              "      <td>240</td>\n",
              "      <td>.</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>30.82770</td>\n",
              "      <td>.</td>\n",
              "      <td>DP=43;VDB=0.199299;SGB=-0.511536;RPB=0.314582;...</td>\n",
              "      <td>GT:PL</td>\n",
              "      <td>0/1:67,0,255</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>CP035312.1</td>\n",
              "      <td>293</td>\n",
              "      <td>.</td>\n",
              "      <td>C</td>\n",
              "      <td>A</td>\n",
              "      <td>6.01423</td>\n",
              "      <td>.</td>\n",
              "      <td>DP=48;VDB=0.440791;SGB=-0.511536;RPB=0.862578;...</td>\n",
              "      <td>GT:PL</td>\n",
              "      <td>0/1:42,0,255</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       #CHROM  POS ID REF ALT      QUAL FILTER  \\\n",
              "0  CP035312.1   90  .   G   T   3.82562      .   \n",
              "1  CP035312.1  141  .   C   A  30.71780      .   \n",
              "2  CP035312.1  215  .   C   A   3.93785      .   \n",
              "3  CP035312.1  240  .   C   A  30.82770      .   \n",
              "4  CP035312.1  293  .   C   A   6.01423      .   \n",
              "\n",
              "                                                INFO FORMAT  \\\n",
              "0  DP=26;VDB=0.42;SGB=-0.453602;RPB=0.235294;MQB=...  GT:PL   \n",
              "1  DP=47;VDB=0.292464;SGB=-0.511536;RPB=0.605083;...  GT:PL   \n",
              "2  DP=44;VDB=0.559209;SGB=-0.511536;RPB=0.958563;...  GT:PL   \n",
              "3  DP=43;VDB=0.199299;SGB=-0.511536;RPB=0.314582;...  GT:PL   \n",
              "4  DP=48;VDB=0.440791;SGB=-0.511536;RPB=0.862578;...  GT:PL   \n",
              "\n",
              "  SRR5470223.sorted.bam  \n",
              "0          0/1:38,0,255  \n",
              "1          0/1:67,0,255  \n",
              "2          0/1:39,0,255  \n",
              "3          0/1:67,0,255  \n",
              "4          0/1:42,0,255  "
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 40
        }
      ]
    },
    {
      "metadata": {
        "id": "z2jTJES_rma0",
        "colab_type": "code",
        "colab": {}
      },
      "cell_type": "code",
      "source": [
        "df_snp.to_excel('snp.xlsx')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "metadata": {
        "id": "1PwAwdqJro8Z",
        "colab_type": "code",
        "outputId": "e5ecabe2-d9cf-4dc8-99c3-446d6f46cb29",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 87
        }
      },
      "cell_type": "code",
      "source": [
        "ls"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "snp.xlsx               SRR5470223.bam         SRR5470223.sra\n",
            "SRP071789.txt          SRR5470223.g.vcf.gz    SRR5470223.vcf.gz.mv.vcf\n",
            "SRR5470223_1.fastq.gz  SRR5470223.sam\n",
            "SRR5470223_2.fastq.gz  SRR5470223.sorted.bam\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "metadata": {
        "id": "_rsmzGWsrwkb",
        "colab_type": "text"
      },
      "cell_type": "markdown",
      "source": [
        "-- one sample에 대한 vcf 파일 만들 수 있게 됨 --"
      ]
    }
  ]
}