{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"mid_test_190416.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"jwosDP6fy7gj","colab_type":"text"},"source":["Data preparation\n","1. Download SRRID : SRR8892235 and convert the sra file into fastq (paired end)\n","2. Map the fastq to reference genome (ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz)\n","3. Do paired-end mapping correction \n","4. Do sort the alignment file\n","5. Call variations (SNP/INDEL) from BAM file <br />\n","*Hint : “bcftools call -mv -Ov -o” \n","\n","Missions \n","1. Convert VCF file to excel file preserving headers (outfilename : SRR8892235.xlsx) \n","2. Filter the VCF file with DP > 20 and convert it into excel (outfilename : SRR8892235.DP20.xlsx)\n","3. Extract genotypes from DP filtered VCF such as AA, TT, GT, TA and so on and save it into excel (outfilename : SRR8892235.DP20.GT.xlsx) \n"]},{"cell_type":"code","metadata":{"id":"0lA1-uYxrNIQ","colab_type":"code","colab":{}},"source":["from google.colab import drive"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"bMW4AOo0rUgF","colab_type":"code","outputId":"4246a583-650f-4d07-f2be-fe4c7ee503a2","executionInfo":{"status":"ok","timestamp":1555376798781,"user_tz":-540,"elapsed":62503,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":131}},"source":["drive.mount('/content/gdrive')"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code\n","\n","Enter your authorization code:\n","··········\n","Mounted at /content/gdrive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"oFPF9NYNrYff","colab_type":"code","outputId":"e7aeae88-9425-4bf7-e917-c7faae657f4c","executionInfo":{"status":"ok","timestamp":1555376882013,"user_tz":-540,"elapsed":969,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":36}},"source":["cd gdrive/My\\ Drive/hbi2019"],"execution_count":0,"outputs":[{"output_type":"stream","text":["/content/gdrive/My Drive/hbi2019\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"iKMIbiN2stzG","colab_type":"code","outputId":"820120b9-325c-4ce2-a4a8-7f7b821d41da","executionInfo":{"status":"ok","timestamp":1555377159375,"user_tz":-540,"elapsed":70052,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":7037}},"source":["! apt-get update \n","! apt install bwa\n","! apt install samtools\n","! apt install bcftools\n","! apt install sra-toolkit"],"execution_count":0,"outputs":[{"output_type":"stream","text":["\r0% [Working]\r            \rIgn:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease\n","\r0% [Connecting to archive.ubuntu.com (91.189.88.149)] [Waiting for headers] [Wa\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease [3,609 B]\n","\r0% [Connecting to archive.ubuntu.com (91.189.88.149)] [Waiting for headers] [2 \r0% [Connecting to archive.ubuntu.com (91.189.88.149)] [Waiting for headers] [Wa\r0% [2 InRelease gpgv 3,609 B] [Connecting to archive.ubuntu.com (91.189.88.149)\r                                                                               \rGet:3 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n","\r0% [2 InRelease gpgv 3,609 B] [Waiting for headers] [3 InRelease 14.2 kB/88.7 k\r                                                                               \rIgn:4 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n","\r0% [2 InRelease gpgv 3,609 B] [Waiting for headers] [3 InRelease 14.2 kB/88.7 k\r                                                                               \rHit:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release\n","\r0% [2 InRelease gpgv 3,609 B] [Waiting for headers] [3 InRelease 14.2 kB/88.7 k\r                                                                               \rHit:6 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n","\r0% [2 InRelease gpgv 3,609 B] [Waiting for headers] [3 InRelease 14.2 kB/88.7 k\r                                                                               \rHit:7 http://archive.ubuntu.com/ubuntu bionic InRelease\n","\r0% [2 InRelease gpgv 3,609 B] [3 InRelease 43.1 kB/88.7 kB 49%] [Waiting for he\r                                                                               \r0% [2 InRelease gpgv 3,609 B] [Waiting for headers] [Waiting for headers]\r                                                                         \r0% [Waiting for headers] [Waiting for headers]\r                                              \rGet:8 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n","\r                                              \rHit:9 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n","Get:10 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ Packages [60.4 kB]\n","Get:12 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic InRelease [15.4 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]\n","Get:15 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [379 kB]\n","Get:16 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [3,915 B]\n","Get:17 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [163 kB]\n","Get:18 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic/main Sources [1,636 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [968 kB]\n","Get:20 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [751 kB]\n","Get:21 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [6,968 B]\n","Get:22 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic/main amd64 Packages [786 kB]\n","Fetched 5,025 kB in 4s (1,170 kB/s)\n","Reading package lists... Done\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-410\n","Use 'apt autoremove' to remove it.\n","Suggested packages:\n","  samtools\n","The following NEW packages will be installed:\n","  bwa\n","0 upgraded, 1 newly installed, 0 to remove and 31 not upgraded.\n","Need to get 176 kB of archives.\n","After this operation, 404 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bwa amd64 0.7.17-1 [176 kB]\n","Fetched 176 kB in 1s (216 kB/s)\n","Selecting previously unselected package bwa.\n","(Reading database ... 131304 files and directories currently installed.)\n","Preparing to unpack .../bwa_0.7.17-1_amd64.deb ...\n","Unpacking bwa (0.7.17-1) ...\n","Setting up bwa (0.7.17-1) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-410\n","Use 'apt autoremove' to remove it.\n","The following additional packages will be installed:\n","  cwltool libc-ares2 libhts2 libhttp-parser2.7.1 libjs-bootstrap nodejs\n","  nodejs-doc python-asn1crypto python-avro python-cachecontrol python-certifi\n","  python-cffi-backend python-chardet python-cryptography python-enum34\n","  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n","  python-mistune python-openssl python-pkg-resources python-pyparsing\n","  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n","  python-schema-salad python-shellescape python-six python-sparqlwrapper\n","  python-typing python-urllib3 python-webencodings\n","Suggested packages:\n","  python-cryptography-doc python-cryptography-vectors python-enum34-doc\n","  python-genshi python-lockfile-doc python-openssl-doc python-openssl-dbg\n","  python-setuptools python-pyparsing-doc python-rdflib-doc python-rdflib-tools\n","  python-socks python-ntlm\n","The following NEW packages will be installed:\n","  cwltool libc-ares2 libhts2 libhttp-parser2.7.1 libjs-bootstrap nodejs\n","  nodejs-doc python-asn1crypto python-avro python-cachecontrol python-certifi\n","  python-cffi-backend python-chardet python-cryptography python-enum34\n","  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n","  python-mistune python-openssl python-pkg-resources python-pyparsing\n","  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n","  python-schema-salad python-shellescape python-six python-sparqlwrapper\n","  python-typing python-urllib3 python-webencodings samtools\n","0 upgraded, 36 newly installed, 0 to remove and 31 not upgraded.\n","Need to get 8,533 kB of archives.\n","After this operation, 40.8 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-urllib3 all 1.22-1 [85.1 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n","Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n","Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n","Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n","Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n","Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n","Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n","Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n","Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n","Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n","Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n","Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n","Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.2 [221 kB]\n","Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n","Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n","Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n","Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n","Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n","Get:34 http://archive.ubuntu.com/ubuntu bionic/main amd64 libc-ares2 amd64 1.14.0-1 [37.1 kB]\n","Get:35 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-parser2.7.1 amd64 2.7.1-2 [20.6 kB]\n","Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n","Fetched 8,533 kB in 2s (3,828 kB/s)\n","Extracting templates from packages: 100%\n","Selecting previously unselected package python-pkg-resources.\n","(Reading database ... 131321 files and directories currently installed.)\n","Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n","Unpacking python-pkg-resources (39.0.1-2) ...\n","Selecting previously unselected package python-six.\n","Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n","Unpacking python-six (1.11.0-2) ...\n","Selecting previously unselected package python-isodate.\n","Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n","Unpacking python-isodate (0.6.0-1) ...\n","Selecting previously unselected package python-pyparsing.\n","Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n","Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n","Selecting previously unselected package python-rdflib.\n","Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n","Unpacking python-rdflib (4.2.1-2) ...\n","Selecting previously unselected package python-certifi.\n","Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n","Unpacking python-certifi (2018.1.18-2) ...\n","Selecting previously unselected package python-chardet.\n","Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n","Unpacking python-chardet (3.0.4-1) ...\n","Selecting previously unselected package python-idna.\n","Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n","Unpacking python-idna (2.6-1) ...\n","Selecting previously unselected package python-urllib3.\n","Preparing to unpack .../08-python-urllib3_1.22-1_all.deb ...\n","Unpacking python-urllib3 (1.22-1) ...\n","Selecting previously unselected package python-requests.\n","Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n","Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n","Selecting previously unselected package python-ruamel.yaml.\n","Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n","Unpacking python-ruamel.yaml (0.15.34-1) ...\n","Selecting previously unselected package python-cachecontrol.\n","Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n","Unpacking python-cachecontrol (0.11.7-1) ...\n","Selecting previously unselected package python-lockfile.\n","Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n","Unpacking python-lockfile (1:0.12.2-2) ...\n","Selecting previously unselected package python-mistune.\n","Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n","Unpacking python-mistune (0.8.3-2) ...\n","Selecting previously unselected package python-rdflib-jsonld.\n","Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n","Unpacking python-rdflib-jsonld (0.4.0-3) ...\n","Selecting previously unselected package python-typing.\n","Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n","Unpacking python-typing (3.6.2-1) ...\n","Selecting previously unselected package python-avro.\n","Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n","Unpacking python-avro (1.8.2+dfsg-1) ...\n","Selecting previously unselected package libjs-bootstrap.\n","Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n","Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n","Selecting previously unselected package python-schema-salad.\n","Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n","Unpacking python-schema-salad (2.6.20171201034858-3) ...\n","Selecting previously unselected package python-shellescape.\n","Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n","Unpacking python-shellescape (3.4.1-2) ...\n","Selecting previously unselected package cwltool.\n","Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n","Unpacking cwltool (1.0.20180302231433-1) ...\n","Selecting previously unselected package libhts2:amd64.\n","Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n","Unpacking libhts2:amd64 (1.7-2) ...\n","Selecting previously unselected package nodejs-doc.\n","Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n","Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n","Selecting previously unselected package python-asn1crypto.\n","Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n","Unpacking python-asn1crypto (0.24.0-1) ...\n","Selecting previously unselected package python-cffi-backend.\n","Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n","Unpacking python-cffi-backend (1.11.5-1) ...\n","Selecting previously unselected package python-enum34.\n","Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n","Unpacking python-enum34 (1.1.6-2) ...\n","Selecting previously unselected package python-ipaddress.\n","Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n","Unpacking python-ipaddress (1.0.17-1) ...\n","Selecting previously unselected package python-cryptography.\n","Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.2_amd64.deb ...\n","Unpacking python-cryptography (2.1.4-1ubuntu1.2) ...\n","Selecting previously unselected package python-webencodings.\n","Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n","Unpacking python-webencodings (0.5-2) ...\n","Selecting previously unselected package python-html5lib.\n","Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n","Unpacking python-html5lib (0.999999999-1) ...\n","Selecting previously unselected package python-openssl.\n","Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n","Unpacking python-openssl (17.5.0-1ubuntu1) ...\n","Selecting previously unselected package python-sparqlwrapper.\n","Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n","Unpacking python-sparqlwrapper (1.7.6-2) ...\n","Selecting previously unselected package samtools.\n","Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n","Unpacking samtools (1.7-1) ...\n","Selecting previously unselected package libc-ares2:amd64.\n","Preparing to unpack .../33-libc-ares2_1.14.0-1_amd64.deb ...\n","Unpacking libc-ares2:amd64 (1.14.0-1) ...\n","Selecting previously unselected package libhttp-parser2.7.1:amd64.\n","Preparing to unpack .../34-libhttp-parser2.7.1_2.7.1-2_amd64.deb ...\n","Unpacking libhttp-parser2.7.1:amd64 (2.7.1-2) ...\n","Selecting previously unselected package nodejs.\n","Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n","Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n","Setting up python-idna (2.6-1) ...\n","Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n","Setting up libhttp-parser2.7.1:amd64 (2.7.1-2) ...\n","Setting up python-asn1crypto (0.24.0-1) ...\n","Setting up python-mistune (0.8.3-2) ...\n","Setting up python-certifi (2018.1.18-2) ...\n","Setting up python-avro (1.8.2+dfsg-1) ...\n","Setting up python-pkg-resources (39.0.1-2) ...\n","Setting up python-ruamel.yaml (0.15.34-1) ...\n","Setting up python-cffi-backend (1.11.5-1) ...\n","Setting up python-six (1.11.0-2) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1) ...\n","Setting up python-isodate (0.6.0-1) ...\n","Setting up python-shellescape (3.4.1-2) ...\n","Setting up python-enum34 (1.1.6-2) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n","Setting up python-lockfile (1:0.12.2-2) ...\n","Setting up python-ipaddress (1.0.17-1) ...\n","Setting up libc-ares2:amd64 (1.14.0-1) ...\n","Setting up libhts2:amd64 (1.7-2) ...\n","Setting up python-webencodings (0.5-2) ...\n","Setting up python-typing (3.6.2-1) ...\n","Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n","Setting up python-urllib3 (1.22-1) ...\n","Setting up python-chardet (3.0.4-1) ...\n","Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n","update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n","Setting up python-rdflib (4.2.1-2) ...\n","Setting up python-sparqlwrapper (1.7.6-2) ...\n","Setting up python-rdflib-jsonld (0.4.0-3) ...\n","Setting up python-cryptography (2.1.4-1ubuntu1.2) ...\n","Setting up python-requests (2.18.4-2ubuntu0.1) ...\n","Setting up python-cachecontrol (0.11.7-1) ...\n","Setting up python-schema-salad (2.6.20171201034858-3) ...\n","Setting up python-html5lib (0.999999999-1) ...\n","Setting up samtools (1.7-1) ...\n","Setting up python-openssl (17.5.0-1ubuntu1) ...\n","Setting up cwltool (1.0.20180302231433-1) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1) ...\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-410\n","Use 'apt autoremove' to remove it.\n","Suggested packages:\n","  python-matplotlib texlive-latex-recommended\n","The following NEW packages will be installed:\n","  bcftools\n","0 upgraded, 1 newly installed, 0 to remove and 31 not upgraded.\n","Need to get 503 kB of archives.\n","After this operation, 1,671 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bcftools amd64 1.7-2 [503 kB]\n","Fetched 503 kB in 1s (388 kB/s)\n","Selecting previously unselected package bcftools.\n","(Reading database ... 132765 files and directories currently installed.)\n","Preparing to unpack .../bcftools_1.7-2_amd64.deb ...\n","Unpacking bcftools (1.7-2) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Setting up bcftools (1.7-2) ...\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-410\n","Use 'apt autoremove' to remove it.\n","The following additional packages will be installed:\n","  blends-common libkdf5-2 libmagic-mgc libmagic1 libmbedcrypto1 libmbedtls-dev\n","  libmbedtls10 libmbedx509-0 libncbi-vdb2 libncbi-wvdb2 med-config menu\n","Suggested packages:\n","  blends-doc file libmbedtls-doc menu-l10n gksu | kde-runtime | ktsuss\n","The following NEW packages will be installed:\n","  blends-common libkdf5-2 libmagic-mgc libmagic1 libmbedcrypto1 libmbedtls-dev\n","  libmbedtls10 libmbedx509-0 libncbi-vdb2 libncbi-wvdb2 med-config menu\n","  sra-toolkit\n","0 upgraded, 13 newly installed, 0 to remove and 31 not upgraded.\n","Need to get 4,458 kB of archives.\n","After this operation, 24.8 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic-mgc amd64 1:5.32-2ubuntu0.2 [184 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic1 amd64 1:5.32-2ubuntu0.2 [68.5 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 menu amd64 2.1.47ubuntu2.1 [349 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu bionic/universe amd64 blends-common all 0.6.100ubuntu2 [15.0 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedcrypto1 amd64 2.8.0-1 [140 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedx509-0 amd64 2.8.0-1 [34.0 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedtls10 amd64 2.8.0-1 [66.8 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedtls-dev amd64 2.8.0-1 [382 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libkdf5-2 amd64 2.8.2-2+dfsg-1build1 [13.1 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-vdb2 amd64 2.8.2-2+dfsg-1build1 [1,098 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-wvdb2 amd64 2.8.2-2+dfsg-1build1 [997 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 sra-toolkit amd64 2.8.2-5+dfsg-1 [1,102 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu bionic/universe amd64 med-config all 3.0.1ubuntu1 [9,404 B]\n","Fetched 4,458 kB in 2s (2,331 kB/s)\n","Preconfiguring packages ...\n","Selecting previously unselected package libmagic-mgc.\n","(Reading database ... 132806 files and directories currently installed.)\n","Preparing to unpack .../00-libmagic-mgc_1%3a5.32-2ubuntu0.2_amd64.deb ...\n","Unpacking libmagic-mgc (1:5.32-2ubuntu0.2) ...\n","Selecting previously unselected package libmagic1:amd64.\n","Preparing to unpack .../01-libmagic1_1%3a5.32-2ubuntu0.2_amd64.deb ...\n","Unpacking libmagic1:amd64 (1:5.32-2ubuntu0.2) ...\n","Selecting previously unselected package menu.\n","Preparing to unpack .../02-menu_2.1.47ubuntu2.1_amd64.deb ...\n","Unpacking menu (2.1.47ubuntu2.1) ...\n","Selecting previously unselected package blends-common.\n","Preparing to unpack .../03-blends-common_0.6.100ubuntu2_all.deb ...\n","Unpacking blends-common (0.6.100ubuntu2) ...\n","Selecting previously unselected package libmbedcrypto1:amd64.\n","Preparing to unpack .../04-libmbedcrypto1_2.8.0-1_amd64.deb ...\n","Unpacking libmbedcrypto1:amd64 (2.8.0-1) ...\n","Selecting previously unselected package libmbedx509-0:amd64.\n","Preparing to unpack .../05-libmbedx509-0_2.8.0-1_amd64.deb ...\n","Unpacking libmbedx509-0:amd64 (2.8.0-1) ...\n","Selecting previously unselected package libmbedtls10:amd64.\n","Preparing to unpack .../06-libmbedtls10_2.8.0-1_amd64.deb ...\n","Unpacking libmbedtls10:amd64 (2.8.0-1) ...\n","Selecting previously unselected package libmbedtls-dev:amd64.\n","Preparing to unpack .../07-libmbedtls-dev_2.8.0-1_amd64.deb ...\n","Unpacking libmbedtls-dev:amd64 (2.8.0-1) ...\n","Selecting previously unselected package libkdf5-2.\n","Preparing to unpack .../08-libkdf5-2_2.8.2-2+dfsg-1build1_amd64.deb ...\n","Unpacking libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n","Selecting previously unselected package libncbi-vdb2.\n","Preparing to unpack .../09-libncbi-vdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n","Unpacking libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n","Selecting previously unselected package libncbi-wvdb2.\n","Preparing to unpack .../10-libncbi-wvdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n","Unpacking libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n","Selecting previously unselected package sra-toolkit.\n","Preparing to unpack .../11-sra-toolkit_2.8.2-5+dfsg-1_amd64.deb ...\n","Unpacking sra-toolkit (2.8.2-5+dfsg-1) ...\n","Selecting previously unselected package med-config.\n","Preparing to unpack .../12-med-config_3.0.1ubuntu1_all.deb ...\n","Unpacking med-config (3.0.1ubuntu1) ...\n","Setting up libmbedcrypto1:amd64 (2.8.0-1) ...\n","Setting up menu (2.1.47ubuntu2.1) ...\n","Setting up libmagic-mgc (1:5.32-2ubuntu0.2) ...\n","Setting up libmagic1:amd64 (1:5.32-2ubuntu0.2) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1) ...\n","Setting up libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Setting up libmbedx509-0:amd64 (2.8.0-1) ...\n","Setting up libmbedtls10:amd64 (2.8.0-1) ...\n","Setting up blends-common (0.6.100ubuntu2) ...\n","Setting up libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n","Setting up libmbedtls-dev:amd64 (2.8.0-1) ...\n","Setting up med-config (3.0.1ubuntu1) ...\n","Adding group `med' (GID 109) ...\n","Done.\n","Setting up libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n","Setting up sra-toolkit (2.8.2-5+dfsg-1) ...\n","Processing triggers for menu (2.1.47ubuntu2.1) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1) ...\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"cPf7NNmJsOVG","colab_type":"code","outputId":"12738ba1-a0de-492c-ffd3-2ed9cd6469b4","executionInfo":{"status":"ok","timestamp":1555376978968,"user_tz":-540,"elapsed":1180,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":36}},"source":["cd sra"],"execution_count":0,"outputs":[{"output_type":"stream","text":["/content/gdrive/My Drive/hbi2019/sra\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"mOY6qfgrrt40","colab_type":"code","outputId":"08a3c178-e9f0-4784-a8b7-aad509ac2be3","executionInfo":{"status":"ok","timestamp":1555377016539,"user_tz":-540,"elapsed":17590,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":336}},"source":["query_sra = 'SRR8892235'\n","sra_dir1 = query_sra[0:3]\n","sra_dir2 = query_sra[0:6]\n","sra_dir3 = query_sra \n","sra_file = query_sra+'.sra'\n","print('Downloading',query_sra)\n","!wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/{sra_dir1}/{sra_dir2}/{sra_dir3}/{sra_file}"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Downloading SRR8892235\n","--2019-04-16 01:09:58--  ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR889/SRR8892235/SRR8892235.sra\n","           => ‘SRR8892235.sra’\n","Resolving ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)... 130.14.250.10, 2607:f220:41e:250::11\n","Connecting to ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)|130.14.250.10|:21... connected.\n","Logging in as anonymous ... Logged in!\n","==> SYST ... done.    ==> PWD ... done.\n","==> TYPE I ... done.  ==> CWD (1) /sra/sra-instant/reads/ByRun/sra/SRR/SRR889/SRR8892235 ... done.\n","==> SIZE SRR8892235.sra ... 481204341\n","==> PASV ... done.    ==> RETR SRR8892235.sra ... done.\n","Length: 481204341 (459M) (unauthoritative)\n","\n","SRR8892235.sra      100%[===================>] 458.91M  36.7MB/s    in 14s     \n","\n","2019-04-16 01:10:13 (32.1 MB/s) - ‘SRR8892235.sra’ saved [481204341]\n","\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"NCAR2Vl5sYn9","colab_type":"code","outputId":"8b27d898-ce46-486f-b296-c5a59adb4357","executionInfo":{"status":"ok","timestamp":1555377472188,"user_tz":-540,"elapsed":273039,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":55}},"source":["! fastq-dump  --origfmt -I  --split-files --gzip {query_sra}.sra"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Read 1831164 spots for SRR8892235.sra\n","Written 1831164 spots for SRR8892235.sra\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"UeJ8_DddseI2","colab_type":"code","outputId":"73be62a0-ea1a-47b0-97fe-f17e3a1b3b74","executionInfo":{"status":"ok","timestamp":1555377478620,"user_tz":-540,"elapsed":728,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":36}},"source":["cd ../references"],"execution_count":0,"outputs":[{"output_type":"stream","text":["/content/gdrive/My Drive/hbi2019/references\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"lmDKhvyYti1U","colab_type":"code","outputId":"66273ca5-7135-47b0-ec3e-45f013d3d1a8","executionInfo":{"status":"ok","timestamp":1555377493980,"user_tz":-540,"elapsed":4098,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":317}},"source":["! wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz"],"execution_count":0,"outputs":[{"output_type":"stream","text":["--2019-04-16 01:18:09--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz\n","           => ‘GCF_000005845.2_ASM584v2_genomic.fna.gz’\n","Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.13, 2607:f220:41e:250::12\n","Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.13|:21... connected.\n","Logging in as anonymous ... Logged in!\n","==> SYST ... done.    ==> PWD ... done.\n","==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2 ... done.\n","==> SIZE GCF_000005845.2_ASM584v2_genomic.fna.gz ... 1379902\n","==> PASV ... done.    ==> RETR GCF_000005845.2_ASM584v2_genomic.fna.gz ... done.\n","Length: 1379902 (1.3M) (unauthoritative)\n","\n","GCF_000005845.2_ASM 100%[===================>]   1.32M  2.71MB/s    in 0.5s    \n","\n","2019-04-16 01:18:10 (2.71 MB/s) - ‘GCF_000005845.2_ASM584v2_genomic.fna.gz’ saved [1379902]\n","\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"bBJdTzvbtsFG","colab_type":"code","colab":{}},"source":["! gzip -d GCF_000005845.2_ASM584v2_genomic.fna.gz"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"w4-tUNGbt0U4","colab_type":"code","outputId":"ee6c23a1-5195-4cc9-f3e4-b70efb7e25b0","executionInfo":{"status":"ok","timestamp":1555377535023,"user_tz":-540,"elapsed":6580,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":185}},"source":["! bwa index GCF_000005845.2_ASM584v2_genomic.fna"],"execution_count":0,"outputs":[{"output_type":"stream","text":["[bwa_index] Pack FASTA... 0.06 sec\n","[bwa_index] Construct BWT for the packed sequence...\n","[bwa_index] 3.44 seconds elapse.\n","[bwa_index] Update BWT... 0.04 sec\n","[bwa_index] Pack forward-only FASTA... 0.03 sec\n","[bwa_index] Construct SA from BWT and Occ... 0.63 sec\n","[main] Version: 0.7.17-r1188\n","[main] CMD: bwa index GCF_000005845.2_ASM584v2_genomic.fna\n","[main] Real time: 4.313 sec; CPU: 4.221 sec\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"vKS_yhsQt4cF","colab_type":"code","outputId":"5454b55b-b046-45c2-968e-040cf44280a8","executionInfo":{"status":"ok","timestamp":1555377565760,"user_tz":-540,"elapsed":799,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":36}},"source":["cd ../sra"],"execution_count":0,"outputs":[{"output_type":"stream","text":["/content/gdrive/My Drive/hbi2019/sra\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"bFKStgLNug3L","colab_type":"code","outputId":"1aced72e-4485-466a-efda-e69c6f4aaa2d","executionInfo":{"status":"ok","timestamp":1555378305570,"user_tz":-540,"elapsed":709167,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":17360}},"source":["! bwa mem ../references/GCF_000005845.2_ASM584v2_genomic.fna {query_sra}_1.fastq.gz {query_sra}_2.fastq.gz -o {query_sra}.sam"],"execution_count":0,"outputs":[{"output_type":"stream","text":["[M::bwa_idx_load_from_disk] read 0 ALT contigs\n","[M::process] read 44018 sequences (10000273 bp)...\n","[M::process] read 44140 sequences (10000387 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15824, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 350, 507)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1067)\n","[M::mem_pestat] mean and std.dev: (380.61, 199.75)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1347)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44018 reads in 8.577 CPU sec, 8.349 real sec\n","[M::process] read 44044 sequences (10000126 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15835, 2, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 349, 506)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1064)\n","[M::mem_pestat] mean and std.dev: (379.70, 200.22)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1343)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44140 reads in 8.757 CPU sec, 8.440 real sec\n","[M::process] read 44202 sequences (10000061 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15933, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 342, 498)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1040)\n","[M::mem_pestat] mean and std.dev: (374.10, 193.79)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1311)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44044 reads in 8.684 CPU sec, 8.450 real sec\n","[M::process] read 44144 sequences (10000461 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15770, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (226, 345, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1048)\n","[M::mem_pestat] mean and std.dev: (376.24, 197.15)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1322)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44202 reads in 8.469 CPU sec, 8.226 real sec\n","[M::process] read 44198 sequences (10000365 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15847, 2, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (226, 345, 504)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1060)\n","[M::mem_pestat] mean and std.dev: (376.93, 197.87)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1338)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44144 reads in 8.559 CPU sec, 8.350 real sec\n","[M::process] read 44236 sequences (10000299 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 15973, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 341, 493)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1031)\n","[M::mem_pestat] mean and std.dev: (371.40, 193.74)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1300)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44198 reads in 8.566 CPU sec, 8.274 real sec\n","[M::process] read 44152 sequences (10000304 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15851, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 347, 501)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1053)\n","[M::mem_pestat] mean and std.dev: (376.05, 196.44)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1329)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44236 reads in 8.603 CPU sec, 8.315 real sec\n","[M::process] read 44176 sequences (10000134 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15842, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 345, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1052)\n","[M::mem_pestat] mean and std.dev: (375.52, 196.59)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1328)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44152 reads in 8.275 CPU sec, 8.075 real sec\n","[M::process] read 44068 sequences (10000134 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15956, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (223, 342, 495)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1039)\n","[M::mem_pestat] mean and std.dev: (371.80, 193.38)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1311)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44176 reads in 8.566 CPU sec, 8.251 real sec\n","[M::process] read 44148 sequences (10000259 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15848, 0, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 349, 503)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1055)\n","[M::mem_pestat] mean and std.dev: (378.23, 196.08)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1331)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44068 reads in 8.576 CPU sec, 8.297 real sec\n","[M::process] read 44130 sequences (10000115 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15869, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (228, 347, 504)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1056)\n","[M::mem_pestat] mean and std.dev: (378.51, 197.68)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1332)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44148 reads in 8.386 CPU sec, 8.090 real sec\n","[M::process] read 44150 sequences (10000327 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15723, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 342, 496)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1040)\n","[M::mem_pestat] mean and std.dev: (372.69, 193.73)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1312)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44130 reads in 8.387 CPU sec, 8.169 real sec\n","[M::process] read 44168 sequences (10000432 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15873, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 349, 502)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1052)\n","[M::mem_pestat] mean and std.dev: (377.82, 196.74)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1327)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44150 reads in 8.523 CPU sec, 8.216 real sec\n","[M::process] read 44112 sequences (10000439 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15852, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 344, 501)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1053)\n","[M::mem_pestat] mean and std.dev: (375.37, 195.86)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1329)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44168 reads in 8.502 CPU sec, 8.194 real sec\n","[M::process] read 44192 sequences (10000493 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15779, 0, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 344, 501)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1053)\n","[M::mem_pestat] mean and std.dev: (375.56, 195.95)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1329)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44112 reads in 8.496 CPU sec, 8.199 real sec\n","[M::process] read 44198 sequences (10000343 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15846, 3, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 342, 493)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1031)\n","[M::mem_pestat] mean and std.dev: (371.28, 194.03)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1300)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44192 reads in 8.451 CPU sec, 8.245 real sec\n","[M::process] read 44184 sequences (10000208 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15882, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 344, 499)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1047)\n","[M::mem_pestat] mean and std.dev: (374.72, 196.04)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1321)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44198 reads in 8.453 CPU sec, 8.162 real sec\n","[M::process] read 44120 sequences (10000067 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15936, 2, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (222, 344, 497)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1047)\n","[M::mem_pestat] mean and std.dev: (373.43, 195.07)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1322)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44184 reads in 8.481 CPU sec, 8.175 real sec\n","[M::process] read 44196 sequences (10000372 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15919, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 344, 498)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1044)\n","[M::mem_pestat] mean and std.dev: (375.28, 196.11)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1317)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44120 reads in 8.481 CPU sec, 8.157 real sec\n","[M::process] read 44222 sequences (10000434 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15805, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 347, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1052)\n","[M::mem_pestat] mean and std.dev: (375.09, 195.91)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1328)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44196 reads in 8.381 CPU sec, 8.099 real sec\n","[M::process] read 44162 sequences (10000104 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15904, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 344, 497)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1041)\n","[M::mem_pestat] mean and std.dev: (373.12, 194.79)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1313)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44222 reads in 8.420 CPU sec, 8.159 real sec\n","[M::process] read 44142 sequences (10000144 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15846, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (223, 341, 492)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1030)\n","[M::mem_pestat] mean and std.dev: (370.77, 192.34)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1299)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44162 reads in 8.703 CPU sec, 8.388 real sec\n","[M::process] read 44108 sequences (10000348 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15859, 1, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (228, 346, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1044)\n","[M::mem_pestat] mean and std.dev: (375.71, 193.63)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1316)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44142 reads in 8.792 CPU sec, 8.475 real sec\n","[M::process] read 44132 sequences (10000172 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15900, 4, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 347, 502)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1056)\n","[M::mem_pestat] mean and std.dev: (376.53, 196.26)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1333)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44108 reads in 8.937 CPU sec, 8.604 real sec\n","[M::process] read 44086 sequences (10000346 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15849, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 346, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1046)\n","[M::mem_pestat] mean and std.dev: (375.36, 194.36)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1319)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44132 reads in 8.970 CPU sec, 8.653 real sec\n","[M::process] read 44002 sequences (10000145 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15827, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (226, 350, 505)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1063)\n","[M::mem_pestat] mean and std.dev: (377.89, 195.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1342)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44086 reads in 8.905 CPU sec, 8.564 real sec\n","[M::process] read 44194 sequences (10000108 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15881, 3, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 348, 502)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1052)\n","[M::mem_pestat] mean and std.dev: (376.48, 193.38)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1327)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44002 reads in 9.242 CPU sec, 8.951 real sec\n","[M::process] read 44108 sequences (10000380 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15891, 2, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (226, 344, 495)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1033)\n","[M::mem_pestat] mean and std.dev: (372.64, 192.16)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1302)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44194 reads in 9.561 CPU sec, 9.245 real sec\n","[M::process] read 44034 sequences (10000055 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15826, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (226, 348, 505)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1063)\n","[M::mem_pestat] mean and std.dev: (377.60, 196.50)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1342)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44108 reads in 9.108 CPU sec, 8.907 real sec\n","[M::process] read 44120 sequences (10000317 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15827, 2, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 346, 498)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1040)\n","[M::mem_pestat] mean and std.dev: (375.17, 192.93)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1311)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44034 reads in 8.963 CPU sec, 8.628 real sec\n","[M::process] read 44104 sequences (10000312 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15829, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 346, 494)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1028)\n","[M::mem_pestat] mean and std.dev: (372.55, 190.75)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1295)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44120 reads in 9.169 CPU sec, 8.861 real sec\n","[M::process] read 44018 sequences (10000130 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15931, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 346, 498)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1046)\n","[M::mem_pestat] mean and std.dev: (374.81, 193.83)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1320)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44104 reads in 9.395 CPU sec, 9.163 real sec\n","[M::process] read 44084 sequences (10000259 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15858, 2, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (228, 345, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1044)\n","[M::mem_pestat] mean and std.dev: (375.00, 191.31)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1316)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44018 reads in 8.885 CPU sec, 8.601 real sec\n","[M::process] read 44188 sequences (10000447 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15830, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 346, 495)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1035)\n","[M::mem_pestat] mean and std.dev: (373.80, 192.16)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1305)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44084 reads in 9.002 CPU sec, 8.672 real sec\n","[M::process] read 43994 sequences (10000443 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15899, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (225, 348, 501)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1053)\n","[M::mem_pestat] mean and std.dev: (375.52, 193.72)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1329)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44188 reads in 8.816 CPU sec, 8.539 real sec\n","[M::process] read 43932 sequences (10000207 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15724, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (228, 346, 496)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1032)\n","[M::mem_pestat] mean and std.dev: (373.15, 189.63)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1300)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43994 reads in 8.826 CPU sec, 8.481 real sec\n","[M::process] read 44034 sequences (10000065 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 15886, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 348, 496)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1026)\n","[M::mem_pestat] mean and std.dev: (374.41, 188.64)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1291)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43932 reads in 8.759 CPU sec, 8.438 real sec\n","[M::process] read 43960 sequences (10000222 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15814, 1, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 345, 497)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1037)\n","[M::mem_pestat] mean and std.dev: (373.19, 190.73)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1307)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44034 reads in 8.761 CPU sec, 8.418 real sec\n","[M::process] read 44010 sequences (10000453 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15886, 0, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 344, 490)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1016)\n","[M::mem_pestat] mean and std.dev: (370.09, 185.71)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1279)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43960 reads in 8.724 CPU sec, 8.418 real sec\n","[M::process] read 44028 sequences (10000312 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15864, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 347, 495)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1031)\n","[M::mem_pestat] mean and std.dev: (374.24, 190.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1299)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44010 reads in 8.435 CPU sec, 8.090 real sec\n","[M::process] read 43896 sequences (10000422 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15808, 1, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 344, 496)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1034)\n","[M::mem_pestat] mean and std.dev: (374.18, 191.19)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1303)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 44028 reads in 8.691 CPU sec, 8.355 real sec\n","[M::process] read 43804 sequences (10000057 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15793, 3, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (233, 350, 500)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1034)\n","[M::mem_pestat] mean and std.dev: (376.64, 190.97)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1301)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43896 reads in 8.873 CPU sec, 8.553 real sec\n","[M::process] read 43842 sequences (10000325 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15757, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (233, 357, 522)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1100)\n","[M::mem_pestat] mean and std.dev: (391.23, 206.34)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1389)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43804 reads in 8.785 CPU sec, 8.476 real sec\n","[M::process] read 43782 sequences (10000062 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15778, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 355, 517)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1089)\n","[M::mem_pestat] mean and std.dev: (387.69, 204.45)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1375)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43842 reads in 8.863 CPU sec, 8.553 real sec\n","[M::process] read 43780 sequences (10000383 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15688, 3, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 355, 513)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1077)\n","[M::mem_pestat] mean and std.dev: (385.59, 201.43)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1359)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43782 reads in 8.915 CPU sec, 8.582 real sec\n","[M::process] read 43878 sequences (10000440 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15644, 1, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 360, 524)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1104)\n","[M::mem_pestat] mean and std.dev: (393.42, 207.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1394)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43780 reads in 8.653 CPU sec, 8.442 real sec\n","[M::process] read 43696 sequences (10000401 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15840, 2, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 356, 521)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1095)\n","[M::mem_pestat] mean and std.dev: (389.96, 206.13)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1382)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43878 reads in 8.900 CPU sec, 8.597 real sec\n","[M::process] read 43912 sequences (10000435 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15851, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 354, 509)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1059)\n","[M::mem_pestat] mean and std.dev: (385.35, 199.07)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1334)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43696 reads in 8.981 CPU sec, 8.676 real sec\n","[M::process] read 43906 sequences (10000188 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15666, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 358, 521)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1101)\n","[M::mem_pestat] mean and std.dev: (389.18, 205.22)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1391)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43912 reads in 8.649 CPU sec, 8.345 real sec\n","[M::process] read 43928 sequences (10000109 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 15755, 1, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (230, 354, 519)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1097)\n","[M::mem_pestat] mean and std.dev: (387.96, 206.51)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1386)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43906 reads in 8.806 CPU sec, 8.514 real sec\n","[M::process] read 43862 sequences (10000447 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15789, 1, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 352, 511)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1071)\n","[M::mem_pestat] mean and std.dev: (384.20, 201.79)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1351)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43928 reads in 8.909 CPU sec, 8.605 real sec\n","[M::process] read 43762 sequences (10000429 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15781, 0, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (232, 355, 519)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1093)\n","[M::mem_pestat] mean and std.dev: (389.99, 206.80)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1380)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43862 reads in 9.021 CPU sec, 8.809 real sec\n","[M::process] read 43740 sequences (10000411 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15770, 1, 4)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 359, 518)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1084)\n","[M::mem_pestat] mean and std.dev: (390.39, 203.48)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1367)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43762 reads in 9.078 CPU sec, 8.769 real sec\n","[M::process] read 43788 sequences (10000223 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15733, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 358, 520)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1092)\n","[M::mem_pestat] mean and std.dev: (388.73, 201.68)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1378)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43740 reads in 9.065 CPU sec, 8.752 real sec\n","[M::process] read 43754 sequences (10000225 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15748, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 361, 522)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1098)\n","[M::mem_pestat] mean and std.dev: (392.46, 206.33)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1386)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43788 reads in 8.566 CPU sec, 8.274 real sec\n","[M::process] read 43724 sequences (10000498 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15700, 2, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 358, 518)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1086)\n","[M::mem_pestat] mean and std.dev: (389.29, 204.01)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1370)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43754 reads in 8.701 CPU sec, 8.397 real sec\n","[M::process] read 43846 sequences (10000450 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15666, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (232, 357, 519)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1093)\n","[M::mem_pestat] mean and std.dev: (389.91, 204.13)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1380)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43724 reads in 8.809 CPU sec, 8.530 real sec\n","[M::process] read 43702 sequences (10000045 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15675, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (233, 360, 525)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1109)\n","[M::mem_pestat] mean and std.dev: (391.65, 206.51)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1401)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43846 reads in 8.663 CPU sec, 8.390 real sec\n","[M::process] read 43720 sequences (10000263 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15746, 2, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (236, 360, 521)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1091)\n","[M::mem_pestat] mean and std.dev: (391.87, 205.00)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1376)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43702 reads in 8.710 CPU sec, 8.394 real sec\n","[M::process] read 43786 sequences (10000416 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15707, 2, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (233, 354, 511)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1067)\n","[M::mem_pestat] mean and std.dev: (384.68, 200.13)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1345)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43720 reads in 8.640 CPU sec, 8.360 real sec\n","[M::process] read 43734 sequences (10000295 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15786, 0, 4)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 356, 521)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1095)\n","[M::mem_pestat] mean and std.dev: (389.45, 205.74)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1382)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43786 reads in 9.019 CPU sec, 8.706 real sec\n","[M::process] read 43762 sequences (10000077 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 15695, 1, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 363, 524)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1104)\n","[M::mem_pestat] mean and std.dev: (393.71, 207.19)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1394)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43734 reads in 9.076 CPU sec, 8.771 real sec\n","[M::process] read 43724 sequences (10000301 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15796, 1, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (231, 354, 515)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1083)\n","[M::mem_pestat] mean and std.dev: (386.04, 202.79)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1367)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43762 reads in 8.776 CPU sec, 8.472 real sec\n","[M::process] read 43728 sequences (10000300 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15746, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (236, 362, 528)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1112)\n","[M::mem_pestat] mean and std.dev: (395.56, 209.11)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1404)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43724 reads in 9.092 CPU sec, 8.763 real sec\n","[M::process] read 43628 sequences (10000231 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15672, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 359, 522)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1096)\n","[M::mem_pestat] mean and std.dev: (392.37, 206.89)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1383)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43728 reads in 8.846 CPU sec, 8.528 real sec\n","[M::process] read 43702 sequences (10000048 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15663, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (237, 361, 517)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1077)\n","[M::mem_pestat] mean and std.dev: (391.62, 202.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1357)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43628 reads in 8.916 CPU sec, 8.602 real sec\n","[M::process] read 43756 sequences (10000382 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15784, 1, 4)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (233, 356, 523)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1103)\n","[M::mem_pestat] mean and std.dev: (390.64, 206.54)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1393)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43702 reads in 8.597 CPU sec, 8.264 real sec\n","[M::process] read 43508 sequences (10000051 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15709, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 356, 517)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1081)\n","[M::mem_pestat] mean and std.dev: (390.42, 203.81)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1363)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43756 reads in 9.150 CPU sec, 8.869 real sec\n","[M::process] read 43656 sequences (10000273 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15746, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (238, 361, 521)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1087)\n","[M::mem_pestat] mean and std.dev: (393.51, 203.49)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1370)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43508 reads in 8.856 CPU sec, 8.562 real sec\n","[M::process] read 43702 sequences (10000238 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15813, 2, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (238, 364, 529)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1111)\n","[M::mem_pestat] mean and std.dev: (396.97, 207.70)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1402)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43656 reads in 8.382 CPU sec, 8.178 real sec\n","[M::process] read 43564 sequences (10000023 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15780, 0, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 364, 529)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1117)\n","[M::mem_pestat] mean and std.dev: (396.76, 208.95)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1411)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43702 reads in 8.340 CPU sec, 8.043 real sec\n","[M::process] read 43672 sequences (10000298 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15767, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (237, 359, 520)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1086)\n","[M::mem_pestat] mean and std.dev: (391.17, 202.36)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1369)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43564 reads in 8.546 CPU sec, 8.245 real sec\n","[M::process] read 43594 sequences (10000295 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 15716, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (237, 362, 525)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1101)\n","[M::mem_pestat] mean and std.dev: (394.86, 206.71)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1389)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43672 reads in 8.439 CPU sec, 8.189 real sec\n","[M::process] read 43562 sequences (10000276 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15720, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (239, 362, 523)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1091)\n","[M::mem_pestat] mean and std.dev: (394.85, 204.95)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1375)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43594 reads in 8.519 CPU sec, 8.199 real sec\n","[M::process] read 43606 sequences (10000078 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15675, 1, 3)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (237, 361, 520)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1086)\n","[M::mem_pestat] mean and std.dev: (392.63, 203.21)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1369)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43562 reads in 8.878 CPU sec, 8.588 real sec\n","[M::process] read 43500 sequences (10000434 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15856, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (236, 365, 531)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1121)\n","[M::mem_pestat] mean and std.dev: (397.15, 207.20)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1416)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43606 reads in 9.007 CPU sec, 8.693 real sec\n","[M::process] read 43460 sequences (10000020 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15731, 1, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (240, 366, 530)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1110)\n","[M::mem_pestat] mean and std.dev: (399.44, 206.62)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1400)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43500 reads in 8.905 CPU sec, 8.697 real sec\n","[M::process] read 43638 sequences (10000043 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 15730, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (239, 364, 523)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1091)\n","[M::mem_pestat] mean and std.dev: (394.28, 202.27)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1375)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43460 reads in 8.968 CPU sec, 8.745 real sec\n","[M::process] read 43412 sequences (10000465 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15773, 1, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 364, 529)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1117)\n","[M::mem_pestat] mean and std.dev: (396.89, 207.24)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1411)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43638 reads in 8.911 CPU sec, 8.725 real sec\n","[M::process] read 43286 sequences (10000198 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15645, 2, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (242, 367, 527)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1097)\n","[M::mem_pestat] mean and std.dev: (397.62, 203.62)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1382)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43412 reads in 8.631 CPU sec, 8.421 real sec\n","[M::process] read 43340 sequences (10000159 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15763, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (241, 367, 526)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1096)\n","[M::mem_pestat] mean and std.dev: (398.54, 203.21)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1381)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43286 reads in 8.634 CPU sec, 8.359 real sec\n","[M::process] read 43300 sequences (10000328 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15603, 0, 2)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (241, 366, 531)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1111)\n","[M::mem_pestat] mean and std.dev: (400.77, 208.28)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1401)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43340 reads in 8.559 CPU sec, 8.281 real sec\n","[M::process] read 18748 sequences (4347491 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15604, 0, 1)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (243, 367, 529)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1101)\n","[M::mem_pestat] mean and std.dev: (399.25, 203.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1387)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 43300 reads in 8.441 CPU sec, 8.226 real sec\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 6751, 0, 0)\n","[M::mem_pestat] skip orientation FF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (244, 367, 524)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1084)\n","[M::mem_pestat] mean and std.dev: (397.64, 200.48)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1364)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] skip orientation RR as there are not enough pairs\n","[M::mem_process_seqs] Processed 18748 reads in 3.824 CPU sec, 3.743 real sec\n","[main] Version: 0.7.17-r1188\n","[main] CMD: bwa mem -o SRR8892235.sam ../references/GCF_000005845.2_ASM584v2_genomic.fna SRR8892235_1.fastq.gz SRR8892235_2.fastq.gz\n","[main] Real time: 706.678 sec; CPU: 730.466 sec\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"V4HukEWAuqep","colab_type":"code","colab":{}},"source":["! samtools fixmate -O bam {query_sra}.sam {query_sra}.bam"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"KK9Fn6V-uzN3","colab_type":"code","outputId":"189db608-299e-41f7-9023-cc128458bc54","executionInfo":{"status":"ok","timestamp":1555378689690,"user_tz":-540,"elapsed":144231,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":36}},"source":["! samtools sort -o {query_sra}.sorted.bam  {query_sra}.bam"],"execution_count":0,"outputs":[{"output_type":"stream","text":["[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"RgmOQp72u2ps","colab_type":"code","outputId":"02be0710-4684-410d-e468-14220ad5dd5d","executionInfo":{"status":"ok","timestamp":1555379224478,"user_tz":-540,"elapsed":448810,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":75}},"source":["! bcftools mpileup -Oz -f ../references/GCF_000005845.2_ASM584v2_genomic.fna -o {query_sra}.g.vcf.gz {query_sra}.sorted.bam\n","! bcftools call -mv -Ov -o {query_sra}.vcf.gz.mv.vcf {query_sra}.g.vcf.gz"],"execution_count":0,"outputs":[{"output_type":"stream","text":["[mpileup] 1 samples in 1 input files\n","Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"sMy5L6eZvxeV","colab_type":"code","outputId":"9158fd6f-069a-4595-8070-27dd7fd395e1","executionInfo":{"status":"ok","timestamp":1555379235760,"user_tz":-540,"elapsed":4903,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":372}},"source":["ll"],"execution_count":0,"outputs":[{"output_type":"stream","text":["total 6765291\n","-rw------- 1 root  276775626 Mar 12 14:30 SRR8700955.sra\n","-rw------- 1 root  205232212 Mar 12 14:30 SRR8700956.sra\n","-rw------- 1 root  263671194 Mar 12 14:29 SRR8700959.sra\n","-rw------- 1 root   98145470 Mar 12 14:33 SRR8704599_1.fastq.gz\n","-rw------- 1 root  101567003 Mar 12 14:33 SRR8704599_2.fastq.gz\n","-rw------- 1 root  262943196 Mar 20 02:01 SRR8704599.bam\n","-rw------- 1 root  120545722 Mar 20 02:07 SRR8704599.g.vcf\n","-rw------- 1 root  948338438 Mar 20 01:24 SRR8704599.sam\n","-rw------- 1 root  176258326 Mar 20 02:06 SRR8704599.sorted.bam\n","-rw------- 1 root  149668594 Mar 12 09:15 SRR8704599.sra\n","-rw------- 1 root  285627215 Apr 16 01:17 SRR8892235_1.fastq.gz\n","-rw------- 1 root  302043998 Apr 16 01:17 SRR8892235_2.fastq.gz\n","-rw------- 1 root  660205291 Apr 16 01:35 SRR8892235.bam\n","-rw------- 1 root  115774057 Apr 16 01:46 SRR8892235.g.vcf.gz\n","-rw------- 1 root 2009188967 Apr 16 01:31 SRR8892235.sam\n","-rw------- 1 root  462768064 Apr 16 01:38 SRR8892235.sorted.bam\n","-rw------- 1 root  481204341 Apr 16 01:09 SRR8892235.sra\n","-rw------- 1 root    7695106 Apr 16 01:47 SRR8892235.vcf.gz.mv.vcf\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"KyiAv8N8vzfU","colab_type":"code","colab":{}},"source":["import pandas as pd\n","\n","df_vcf = pd.read_csv('./SRR8892235.vcf.gz.mv.vcf',sep='\\t',comment='#',header=None)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"Gy5DXNWYv3uz","colab_type":"code","colab":{}},"source":["columns = !cat ./SRR8892235.vcf.gz.mv.vcf | grep '#' | tail -1\n","columns = columns[0].split('\\t')\n","df_vcf.columns = columns"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"pvxySArk1F59","colab_type":"code","outputId":"0dba8956-7e64-4a82-ad24-83048c6c2dbb","executionInfo":{"status":"ok","timestamp":1555379286858,"user_tz":-540,"elapsed":1463,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":204}},"source":["df_vcf.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>#CHROM</th>\n","      <th>POS</th>\n","      <th>ID</th>\n","      <th>REF</th>\n","      <th>ALT</th>\n","      <th>QUAL</th>\n","      <th>FILTER</th>\n","      <th>INFO</th>\n","      <th>FORMAT</th>\n","      <th>SRR8892235.sorted.bam</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>NC_000913.3</td>\n","      <td>301</td>\n","      <td>.</td>\n","      <td>CTTTTTTTTT</td>\n","      <td>CTTTTTTTT</td>\n","      <td>15.8866</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=170;IMF=0.982659;DP=173;VDB=0.203432...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:43,73,0</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>NC_000913.3</td>\n","      <td>573</td>\n","      <td>.</td>\n","      <td>C</td>\n","      <td>A</td>\n","      <td>225.0000</td>\n","      <td>.</td>\n","      <td>DP=158;VDB=0.646972;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>NC_000913.3</td>\n","      <td>747</td>\n","      <td>.</td>\n","      <td>A</td>\n","      <td>G</td>\n","      <td>225.0000</td>\n","      <td>.</td>\n","      <td>DP=118;VDB=0.895817;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>NC_000913.3</td>\n","      <td>759</td>\n","      <td>.</td>\n","      <td>C</td>\n","      <td>T</td>\n","      <td>225.0000</td>\n","      <td>.</td>\n","      <td>DP=106;VDB=0.983654;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>NC_000913.3</td>\n","      <td>774</td>\n","      <td>.</td>\n","      <td>T</td>\n","      <td>C</td>\n","      <td>225.0000</td>\n","      <td>.</td>\n","      <td>DP=97;VDB=0.990106;SGB=-0.693147;MQSB=1;MQ0F=0...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["        #CHROM  POS ID         REF        ALT      QUAL FILTER  \\\n","0  NC_000913.3  301  .  CTTTTTTTTT  CTTTTTTTT   15.8866      .   \n","1  NC_000913.3  573  .           C          A  225.0000      .   \n","2  NC_000913.3  747  .           A          G  225.0000      .   \n","3  NC_000913.3  759  .           C          T  225.0000      .   \n","4  NC_000913.3  774  .           T          C  225.0000      .   \n","\n","                                                INFO FORMAT  \\\n","0  INDEL;IDV=170;IMF=0.982659;DP=173;VDB=0.203432...  GT:PL   \n","1  DP=158;VDB=0.646972;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","2  DP=118;VDB=0.895817;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","3  DP=106;VDB=0.983654;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","4  DP=97;VDB=0.990106;SGB=-0.693147;MQSB=1;MQ0F=0...  GT:PL   \n","\n","  SRR8892235.sorted.bam  \n","0           1/1:43,73,0  \n","1         1/1:255,255,0  \n","2         1/1:255,255,0  \n","3         1/1:255,255,0  \n","4         1/1:255,255,0  "]},"metadata":{"tags":[]},"execution_count":32}]},{"cell_type":"code","metadata":{"id":"GNmx8X1swAky","colab_type":"code","colab":{}},"source":["df_vcf.to_excel('SRR8892235.xlsx')"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"GCDECjy3wLZV","colab_type":"code","outputId":"fe8fc471-dbd2-4040-d106-b65ef7f59b5c","executionInfo":{"status":"ok","timestamp":1555379342353,"user_tz":-540,"elapsed":1500,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":204}},"source":["m_indel = df_vcf['INFO'].apply(lambda x : x[0:5] == 'INDEL')\n","df_vcf[m_indel].head() "],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>#CHROM</th>\n","      <th>POS</th>\n","      <th>ID</th>\n","      <th>REF</th>\n","      <th>ALT</th>\n","      <th>QUAL</th>\n","      <th>FILTER</th>\n","      <th>INFO</th>\n","      <th>FORMAT</th>\n","      <th>SRR8892235.sorted.bam</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>NC_000913.3</td>\n","      <td>301</td>\n","      <td>.</td>\n","      <td>CTTTTTTTTT</td>\n","      <td>CTTTTTTTT</td>\n","      <td>15.8866</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=170;IMF=0.982659;DP=173;VDB=0.203432...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:43,73,0</td>\n","    </tr>\n","    <tr>\n","      <th>77</th>\n","      <td>NC_000913.3</td>\n","      <td>5595</td>\n","      <td>.</td>\n","      <td>G</td>\n","      <td>GA</td>\n","      <td>221.0000</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=127;IMF=0.92029;DP=138;VDB=0.876491;...</td>\n","      <td>GT:PL</td>\n","      <td>0/1:255,0,163</td>\n","    </tr>\n","    <tr>\n","      <th>160</th>\n","      <td>NC_000913.3</td>\n","      <td>14404</td>\n","      <td>.</td>\n","      <td>GGGCGGCGGCGGTTTTGGCGGCGGCG</td>\n","      <td>GGGCGGCGGCG</td>\n","      <td>228.0000</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=138;IMF=0.945205;DP=146;VDB=9.08582e...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,173,0</td>\n","    </tr>\n","    <tr>\n","      <th>219</th>\n","      <td>NC_000913.3</td>\n","      <td>20702</td>\n","      <td>.</td>\n","      <td>CT</td>\n","      <td>CTGCAAAAGCCAGCAT</td>\n","      <td>219.0000</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=152;IMF=0.938272;DP=162;VDB=3.33574e...</td>\n","      <td>GT:PL</td>\n","      <td>0/1:255,0,133</td>\n","    </tr>\n","    <tr>\n","      <th>224</th>\n","      <td>NC_000913.3</td>\n","      <td>21367</td>\n","      <td>.</td>\n","      <td>C</td>\n","      <td>CT</td>\n","      <td>218.0000</td>\n","      <td>.</td>\n","      <td>INDEL;IDV=163;IMF=0.91573;DP=178;VDB=0.973195;...</td>\n","      <td>GT:PL</td>\n","      <td>0/1:255,0,46</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["          #CHROM    POS ID                         REF               ALT  \\\n","0    NC_000913.3    301  .                  CTTTTTTTTT         CTTTTTTTT   \n","77   NC_000913.3   5595  .                           G                GA   \n","160  NC_000913.3  14404  .  GGGCGGCGGCGGTTTTGGCGGCGGCG       GGGCGGCGGCG   \n","219  NC_000913.3  20702  .                          CT  CTGCAAAAGCCAGCAT   \n","224  NC_000913.3  21367  .                           C                CT   \n","\n","         QUAL FILTER                                               INFO  \\\n","0     15.8866      .  INDEL;IDV=170;IMF=0.982659;DP=173;VDB=0.203432...   \n","77   221.0000      .  INDEL;IDV=127;IMF=0.92029;DP=138;VDB=0.876491;...   \n","160  228.0000      .  INDEL;IDV=138;IMF=0.945205;DP=146;VDB=9.08582e...   \n","219  219.0000      .  INDEL;IDV=152;IMF=0.938272;DP=162;VDB=3.33574e...   \n","224  218.0000      .  INDEL;IDV=163;IMF=0.91573;DP=178;VDB=0.973195;...   \n","\n","    FORMAT SRR8892235.sorted.bam  \n","0    GT:PL           1/1:43,73,0  \n","77   GT:PL         0/1:255,0,163  \n","160  GT:PL         1/1:255,173,0  \n","219  GT:PL         0/1:255,0,133  \n","224  GT:PL          0/1:255,0,46  "]},"metadata":{"tags":[]},"execution_count":35}]},{"cell_type":"code","metadata":{"id":"HERc54f_xPa0","colab_type":"code","colab":{}},"source":["df_vcf_snp = df_vcf[~m_indel] "],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"G9mWuGgMxRu2","colab_type":"code","outputId":"3da9d4d1-88b6-4ef6-b540-0cd0a9a8eb90","executionInfo":{"status":"ok","timestamp":1555379378333,"user_tz":-540,"elapsed":1806,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":35}},"source":["df_vcf_snp['INFO'].apply(lambda x : x.split(';')[0].replace('DP=','')).values.astype(int)"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["array([158, 118, 106, ..., 117, 115, 115])"]},"metadata":{"tags":[]},"execution_count":38}]},{"cell_type":"code","metadata":{"id":"y5prwOB1xXbq","colab_type":"code","colab":{}},"source":["dp_dist = df_vcf_snp['INFO'].apply(lambda x : x.split(';')[0].replace('DP=','')).values.astype(int)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"l7-58sn_xaMj","colab_type":"code","colab":{}},"source":["dp_dist = pd.Series(dp_dist)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"BDDJWaZUxxmS","colab_type":"code","colab":{}},"source":["m_dp = dp_dist.values > 20"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"WhSVnWarx2DH","colab_type":"code","colab":{}},"source":["df_vcf_snp_highdp = df_vcf_snp[m_dp]"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"i0EQqnmn12G8","colab_type":"code","outputId":"3e2be683-dcbd-4168-fb87-d204d8d0c61a","executionInfo":{"status":"ok","timestamp":1555379498075,"user_tz":-540,"elapsed":1465,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":35}},"source":["df_vcf.shape, df_vcf_snp.shape, df_vcf_snp_highdp.shape"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["((57146, 10), (56726, 10), (55561, 10))"]},"metadata":{"tags":[]},"execution_count":46}]},{"cell_type":"code","metadata":{"id":"luc6v5WTx3T5","colab_type":"code","colab":{}},"source":["df_vcf_snp_highdp.to_excel('SRR8892235.DP20.xlsx')"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"MI-7Gd540hwa","colab_type":"code","outputId":"401273a9-9d50-4b0a-f4ab-9f2adbcb8ee5","executionInfo":{"status":"ok","timestamp":1555379570109,"user_tz":-540,"elapsed":1266,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":204}},"source":["df_vcf_snp_highdp.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>#CHROM</th>\n","      <th>POS</th>\n","      <th>ID</th>\n","      <th>REF</th>\n","      <th>ALT</th>\n","      <th>QUAL</th>\n","      <th>FILTER</th>\n","      <th>INFO</th>\n","      <th>FORMAT</th>\n","      <th>SRR8892235.sorted.bam</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>1</th>\n","      <td>NC_000913.3</td>\n","      <td>573</td>\n","      <td>.</td>\n","      <td>C</td>\n","      <td>A</td>\n","      <td>225.0</td>\n","      <td>.</td>\n","      <td>DP=158;VDB=0.646972;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>NC_000913.3</td>\n","      <td>747</td>\n","      <td>.</td>\n","      <td>A</td>\n","      <td>G</td>\n","      <td>225.0</td>\n","      <td>.</td>\n","      <td>DP=118;VDB=0.895817;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>NC_000913.3</td>\n","      <td>759</td>\n","      <td>.</td>\n","      <td>C</td>\n","      <td>T</td>\n","      <td>225.0</td>\n","      <td>.</td>\n","      <td>DP=106;VDB=0.983654;SGB=-0.693147;MQSB=1;MQ0F=...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>NC_000913.3</td>\n","      <td>774</td>\n","      <td>.</td>\n","      <td>T</td>\n","      <td>C</td>\n","      <td>225.0</td>\n","      <td>.</td>\n","      <td>DP=97;VDB=0.990106;SGB=-0.693147;MQSB=1;MQ0F=0...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,255,0</td>\n","    </tr>\n","    <tr>\n","      <th>5</th>\n","      <td>NC_000913.3</td>\n","      <td>798</td>\n","      <td>.</td>\n","      <td>G</td>\n","      <td>A</td>\n","      <td>228.0</td>\n","      <td>.</td>\n","      <td>DP=94;VDB=0.957278;SGB=-0.693147;RPB=1;MQB=1;M...</td>\n","      <td>GT:PL</td>\n","      <td>1/1:255,228,0</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["        #CHROM  POS ID REF ALT   QUAL FILTER  \\\n","1  NC_000913.3  573  .   C   A  225.0      .   \n","2  NC_000913.3  747  .   A   G  225.0      .   \n","3  NC_000913.3  759  .   C   T  225.0      .   \n","4  NC_000913.3  774  .   T   C  225.0      .   \n","5  NC_000913.3  798  .   G   A  228.0      .   \n","\n","                                                INFO FORMAT  \\\n","1  DP=158;VDB=0.646972;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","2  DP=118;VDB=0.895817;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","3  DP=106;VDB=0.983654;SGB=-0.693147;MQSB=1;MQ0F=...  GT:PL   \n","4  DP=97;VDB=0.990106;SGB=-0.693147;MQSB=1;MQ0F=0...  GT:PL   \n","5  DP=94;VDB=0.957278;SGB=-0.693147;RPB=1;MQB=1;M...  GT:PL   \n","\n","  SRR8892235.sorted.bam  \n","1         1/1:255,255,0  \n","2         1/1:255,255,0  \n","3         1/1:255,255,0  \n","4         1/1:255,255,0  \n","5         1/1:255,228,0  "]},"metadata":{"tags":[]},"execution_count":49}]},{"cell_type":"code","metadata":{"id":"xrudSeggy09E","colab_type":"code","colab":{}},"source":["df_vcf_snp_highdp.columns = columns\n","c_accessions = columns[9:]"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"h0Jg_QiDx9ZO","colab_type":"code","colab":{}},"source":["import numpy as np\n","def call_gt(x):\n","  ref = x['REF']\n","  alt = x['ALT'].split(',')\n","  gts = np.array([ref]+alt)\n","  calls = x[c_accessions]\n","  gts_list = []\n","  for call in calls :\n","    gt = call.split(':')[0]\n","    ixs = np.array(gt.split('/')).astype(int) \n","    gts_list.append(''.join(gts[ixs]))\n","  return gts_list"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"jCRXToqUzXgI","colab_type":"code","colab":{}},"source":["data = df_vcf_snp_highdp.apply(call_gt, axis=1) "],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"2lNZo05e3Xy4","colab_type":"code","colab":{}},"source":["data = np.stack(data)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"53q_bBMp3Y38","colab_type":"code","outputId":"f3a883bc-22e0-47be-c2aa-d22a60c9570f","executionInfo":{"status":"ok","timestamp":1555379898946,"user_tz":-540,"elapsed":556,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":35}},"source":["data.shape"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["(55561, 1)"]},"metadata":{"tags":[]},"execution_count":59}]},{"cell_type":"code","metadata":{"id":"WhzFbpdS3ZzX","colab_type":"code","colab":{}},"source":["df_gt_call = pd.DataFrame(data=data)"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"J3phJZyc3oI1","colab_type":"code","colab":{}},"source":["df_gt_call.columns = [x.split('.')[0] for x in c_accessions]"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"mXYenUC53mGO","colab_type":"code","outputId":"c12b3a0c-e095-4330-f6c0-7cb612f56a25","executionInfo":{"status":"ok","timestamp":1555379990118,"user_tz":-540,"elapsed":776,"user":{"displayName":"HaLim Park","photoUrl":"","userId":"09983875586764085696"}},"colab":{"base_uri":"https://localhost:8080/","height":204}},"source":["df_gt_call.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>SRR8892235</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>AA</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>GG</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>TT</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>CC</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>AA</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["  SRR8892235\n","0         AA\n","1         GG\n","2         TT\n","3         CC\n","4         AA"]},"metadata":{"tags":[]},"execution_count":66}]},{"cell_type":"code","metadata":{"id":"T3HY1en43qhP","colab_type":"code","colab":{}},"source":["df_gt_call.to_excel('SRR8892235.DP20.GT.xlsx')"],"execution_count":0,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"aG0ju7Zqckgk","colab_type":"text"},"source":["---"]}]}